<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="lecook" />

<meta name="date" content="2022-02-23" />

<title>Dunnart ChIP-seq data pre-processing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">chipseq_cross_species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lecook/chipseq-cross-species">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Dunnart ChIP-seq data pre-processing</h1>
<h4 class="author">lecook</h4>
<h4 class="date">2022-02-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chipseq-cross-species/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220209code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220209)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220209code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220209)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestreebe43c257aafcb4287b3194e8f6da73cab02f22batargetblankbe43c25a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lecook/chipseq-cross-species/tree/be43c257aafcb4287b3194e8f6da73cab02f22ba" target="_blank">be43c25</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestreebe43c257aafcb4287b3194e8f6da73cab02f22batargetblankbe43c25a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lecook/chipseq-cross-species/tree/be43c257aafcb4287b3194e8f6da73cab02f22ba" target="_blank">be43c25</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    data/genomic_data/
    Ignored:    output/annotations/
    Ignored:    output/bam_files/
    Ignored:    output/filtered_peaks/
    Ignored:    output/logs/
    Ignored:    output/peaks/
    Ignored:    output/plots/
    Ignored:    output/qc/A-1_input.SeqDepthNorm.bw
    Ignored:    output/qc/A-1_input.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-1_input.ccurve.txt
    Ignored:    output/qc/A-1_input.extrap.txt
    Ignored:    output/qc/A-1_input_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-1_input_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-2_H3K4me3.SeqDepthNorm.bw
    Ignored:    output/qc/A-2_H3K4me3.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-2_H3K4me3.ccurve.txt
    Ignored:    output/qc/A-2_H3K4me3.extrap.txt
    Ignored:    output/qc/A-2_H3K4me3_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-2_H3K4me3_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/A-3_H3K27ac.SeqDepthNorm.bw
    Ignored:    output/qc/A-3_H3K27ac.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-3_H3K27ac.ccurve.txt
    Ignored:    output/qc/A-3_H3K27ac.extrap.txt
    Ignored:    output/qc/A-3_H3K27ac_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-3_H3K27ac_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/B-1_input.SeqDepthNorm.bw
    Ignored:    output/qc/B-1_input.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-1_input.ccurve.txt
    Ignored:    output/qc/B-1_input.extrap.txt
    Ignored:    output/qc/B-1_input_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-1_input_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-2_H3K4me3.SeqDepthNorm.bw
    Ignored:    output/qc/B-2_H3K4me3.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-2_H3K4me3.ccurve.txt
    Ignored:    output/qc/B-2_H3K4me3.extrap.txt
    Ignored:    output/qc/B-2_H3K4me3_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-2_H3K4me3_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_A-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/B-3_H3K27ac.SeqDepthNorm.bw
    Ignored:    output/qc/B-3_H3K27ac.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-3_H3K27ac.ccurve.txt
    Ignored:    output/qc/B-3_H3K27ac.extrap.txt
    Ignored:    output/qc/B-3_H3K27ac_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-3_H3K27ac_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/bamPEFragmentSize_rawcounts.tab
    Ignored:    output/qc/bamPEFragmentSize_rawcounts_dunnart_downSampled.tab
    Ignored:    output/qc/multiBAM_fingerprint_metrics.txt
    Ignored:    output/qc/multiBAM_fingerprint_metrics_dunnart_downSampled.txt
    Ignored:    output/qc/multiBAM_fingerprint_rawcounts.txt
    Ignored:    output/qc/multiBAM_fingerprint_rawcounts_dunnart_downSampled.txt
    Ignored:    output/qc/multibamsum.npz
    Ignored:    output/qc/multibamsum.tab
    Ignored:    output/qc/multibamsum_dunnart_downSampled.npz
    Ignored:    output/qc/multibamsum_dunnart_downSampled.tab
    Ignored:    output/qc/pearsoncor_multibamsum_matrix.txt
    Ignored:    output/qc/pearsoncor_multibamsum_matrix_dunnart_downSampled.txt
    Ignored:    output/wga/
    Ignored:    results/

Untracked files:
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  data/raw_reads/
    Untracked:  output/qc/ucsc_alignment/
    Untracked:  output/rnaseq/
    Untracked:  slurm-33931929.out
    Untracked:  slurm-33931930.out
    Untracked:  slurm-33931931.out
    Untracked:  slurm-33931932.out
    Untracked:  slurm-33931933.out
    Untracked:  slurm-33931934.out
    Untracked:  slurm-33934389.out
    Untracked:  slurm-33934390.out
    Untracked:  slurm-33934391.out
    Untracked:  slurm-33934392.out
    Untracked:  slurm-33934393.out
    Untracked:  slurm-33934394.out
    Untracked:  slurm-33935177.out
    Untracked:  slurm-33935178.out
    Untracked:  slurm-33935179.out
    Untracked:  slurm-33935180.out
    Untracked:  slurm-33935181.out
    Untracked:  slurm-33935182.out
    Untracked:  slurm-33938210.out
    Untracked:  slurm-33938211.out
    Untracked:  slurm-33938212.out
    Untracked:  slurm-33938213.out
    Untracked:  slurm-33938214.out
    Untracked:  slurm-33938215.out
    Untracked:  slurm-33948861.out
    Untracked:  slurm-33948862.out
    Untracked:  slurm-33948863.out
    Untracked:  slurm-33948864.out
    Untracked:  slurm-33948865.out
    Untracked:  slurm-33948866.out
    Untracked:  slurm-33950657.out
    Untracked:  slurm-33950658.out
    Untracked:  slurm-33950659.out
    Untracked:  slurm-33950660.out
    Untracked:  slurm-33950661.out
    Untracked:  slurm-33950662.out
    Untracked:  slurm-33950663.out
    Untracked:  slurm-33950664.out
    Untracked:  slurm-33950810.out
    Untracked:  slurm-33950814.out
    Untracked:  slurm-33950815.out
    Untracked:  slurm-33950822.out
    Untracked:  slurm-33952537.out
    Untracked:  slurm-33952538.out
    Untracked:  slurm-33952553.out

Unstaged changes:
    Modified:   analysis/peak_level_comparisons.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/dunnart_data_processing.Rmd</code>) and HTML (<code>docs/dunnart_data_processing.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/2103ae7d50dd723698bf11af8da5e04882dcda03/analysis/dunnart_data_processing.Rmd" target="_blank">2103ae7</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/802fa8c2b58dc997fd35e53ad4b7878bcda64fa4/analysis/dunnart_data_processing.Rmd" target="_blank">802fa8c</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/ac21f8a403b8aac648497516dc7c66752d57bd5a/docs/dunnart_data_processing.html" target="_blank">ac21f8a</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/4e6c7f00b5c7d5967bba0e6abdf549c3cc4f1643/analysis/dunnart_data_processing.Rmd" target="_blank">4e6c7f0</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/42e48bec4da69be2a8cb33e711fc29ef147f1a18/docs/dunnart_data_processing.html" target="_blank">42e48be</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/493419de6590d047f056bd949fb2c9ec546d0093/analysis/dunnart_data_processing.Rmd" target="_blank">493419d</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/07816acb8201e6681c9e3eb22cee5968751c6d0c/docs/dunnart_data_processing.html" target="_blank">07816ac</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/e68ae60de8b3208b46dff81fdfd594fb5cfa2b17/analysis/dunnart_data_processing.Rmd" target="_blank">e68ae60</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
add module load bedtools
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/837da64a635c4a7d77ef75bc8e374cbd9f540904/docs/dunnart_data_processing.html" target="_blank">837da64</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/1b9dec301ee21b74c35a32f0e87c2bd1cc3449b1/analysis/dunnart_data_processing.Rmd" target="_blank">1b9dec3</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
fix file names
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/11afbc76c9b32c1f56b3b94182013c0660e027bc/docs/dunnart_data_processing.html" target="_blank">11afbc7</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/e3431cb0ea7062851539bff5a015c5f453199d7f/analysis/dunnart_data_processing.Rmd" target="_blank">e3431cb</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/5ebf4b5ab2dbc48eac218be1f666efce5c2205a3/docs/dunnart_data_processing.html" target="_blank">5ebf4b5</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/32d3f2d95de3e2eeb5def086df0b7ebd9f527db0/analysis/dunnart_data_processing.Rmd" target="_blank">32d3f2d</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/98edd22c30bfd4328997ea92d5e9c5fe68a5234c/docs/dunnart_data_processing.html" target="_blank">98edd22</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/537629e718b52c6eccbf102238edc1bbd4b73b46/analysis/dunnart_data_processing.Rmd" target="_blank">537629e</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
erroneous brackets
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/c2dc26d68e6a10e01c552c707c0880f424e589f5/analysis/dunnart_data_processing.Rmd" target="_blank">c2dc26d</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
update directories
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/4176dacb7a81b2d6bb678db67969cec831995db0/docs/dunnart_data_processing.html" target="_blank">4176dac</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/323d8b6a03950741027f0b6329a51b8f0c31d796/analysis/dunnart_data_processing.Rmd" target="_blank">323d8b6</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
copied scripts over
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/ae523af3e11ade3cc2e0c7884c96bfdc5faa66b1/analysis/dunnart_data_processing.Rmd" target="_blank">ae523af</a>
</td>
<td>
lecook
</td>
<td>
2022-03-02
</td>
<td>
fix code formatting for bam coverage rule
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/9d9a99533caf5d7803ed5576afd8786ef296a7ea/analysis/dunnart_data_processing.Rmd" target="_blank">9d9a995</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
udpated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/275d2299013a6db02d57b94c22d287788ec16d15/docs/dunnart_data_processing.html" target="_blank">275d229</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/e69f5512379cb1185f85dff6bded43ddca1237d7/analysis/dunnart_data_processing.Rmd" target="_blank">e69f551</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5fb3661d2eb6bfa7968354bd435070ebf80e7a21/analysis/dunnart_data_processing.Rmd" target="_blank">5fb3661</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
clean up
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/9aa0925728dd7ad1d6fb64dfff484bc6794739c0/analysis/dunnart_data_processing.Rmd" target="_blank">9aa0925</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/277f9e450a12bc300ab2677db8b2640355fb7c53/docs/dunnart_data_processing.html" target="_blank">277f9e4</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/072d4e4a258999fb4b217a6e9b063ed950504bf3/docs/dunnart_data_processing.html" target="_blank">072d4e4</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/fc2805319d09230ecf9d007aa69d51fc34069b27/docs/dunnart_data_processing.html" target="_blank">fc28053</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/08852744cc90c2fb59845baab99e011c767fbe5c/docs/dunnart_data_processing.html" target="_blank">0885274</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/ba5cb47f8f03435189678977b388ad37040c35f3/docs/dunnart_data_processing.html" target="_blank">ba5cb47</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
build html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/369a400fe152f78f6e805fdc2cb53fa491ef6b74/analysis/dunnart_data_processing.Rmd" target="_blank">369a400</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/ce570a13c90367373965502a716749fc3ea8f22d/analysis/dunnart_data_processing.Rmd" target="_blank">ce570a1</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
removed non-dedup index bam from rule all
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5882c3506b6138e3e4b9babfa2fa493435e89393/analysis/dunnart_data_processing.Rmd" target="_blank">5882c35</a>
</td>
<td>
lecook
</td>
<td>
2022-02-25
</td>
<td>
test
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/1fa92d9e841739434d4e6cffcca1d57dfebcf651/analysis/dunnart_data_processing.Rmd" target="_blank">1fa92d9</a>
</td>
<td>
lecook
</td>
<td>
2022-02-23
</td>
<td>
add analysis files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="snakemake-pipeline" class="section level1">
<h1>Snakemake pipeline</h1>
<p>For full snakemake script see code/dunnart_peak_calling/dunnart_snakefile.</p>
<div id="create-conda-environment" class="section level2">
<h2>Create conda environment</h2>
<pre class="bash"><code># create conda environment with all packages and dependencies
conda create --name chip bowtie2 samtools picard deeptools multiqc phantompeakqualtools preseq macs2 bedtools fastqc snakemake numpy pybedtools

# save environment to a yaml file
conda env export &gt; chip_environment.yml

# activate environment before running pipeline
conda activate chip</code></pre>
</div>
<div id="run-snakemake-from-command-line" class="section level2">
<h2>Run snakemake from command line</h2>
<p>Cluster configuration file: code/configs/cluster.json</p>
<p>Sample configuration file: code/configs/config.yaml</p>
<p>Sample text file: code/configs/SSR.text</p>
<pre class="bash"><code>module load snakemake/6.6.1
snakemake --snakemakefile code/dunnart_peak_calling/dunnart_snakefile -j 6 --cluster-config envs/cluster.json --cluster &quot;sbatch -A {cluster.account} -t {cluster.time} -p {cluster.partition} --nodes {cluster.nodes} --ntasks {cluster.ntasks} --mem {cluster.mem}&quot; &amp;</code></pre>
</div>
<div id="effective-genome-length" class="section level2">
<h2>Effective genome length</h2>
<p>We can approximate effective genome size for various read lengths using the khmer program and <code>unique-kmers.py</code>. This will estimate the number of unique kmers (for a specified length kmer) which can be used to infer the total uniquely mappable genome. (I.e it doesn’t include highly repetitive regions). <a href="https://khmer.readthedocs.io/en/v2.1.1/user/scripts.html" class="uri">https://khmer.readthedocs.io/en/v2.1.1/user/scripts.html</a></p>
<p>This was a suggestion of deepTools: <a href="https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html" class="uri">https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html</a></p>
<pre class="bash"><code>module load pip/21.2.4-python-3.8.6
pip2.7 install khmer</code></pre>
<p>Run <code>unique-kmers.py</code> on dunnart genome for read length of 150bp:</p>
<pre class="bash"><code>/usr/local/bin/unique-kmers.py -k 150 Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa</code></pre>
<pre><code>Estimated number of unique 150-mers in /Users/lauracook/../../Volumes/macOS/genomes/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fasta: 2740338543
Total estimated number of unique 150-mers: 2740338543</code></pre>
</div>
<div id="indexing-genome-file" class="section level2">
<h2>Indexing genome file</h2>
<p>Load modules:</p>
<pre class="bash"><code>module load gcc/8.3.0
module load bowtie2/2.3.5.1</code></pre>
<p>Build index:</p>
<pre class="bash"><code>bowtie2-build Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr</code></pre>
</div>
<div id="align-and-filter-dunnart-reads" class="section level2">
<h2>Align and filter dunnart reads</h2>
<p>Uses samtools and picards MarkDuplicates to remove low quality reads. This is based on and in accordance with the ENCODE Guidelines and pipeline (<a href="https://github.com/ENCODE-DCC/chip-seq-pipeline2" class="uri">https://github.com/ENCODE-DCC/chip-seq-pipeline2</a>)</p>
<p><strong>Library Complexity ChIP-seq Standards:</strong></p>
<table>
<thead>
<tr class="header">
<th align="center">PBC1</th>
<th align="center">PBC2</th>
<th align="center">Bottlenecking level</th>
<th align="center">NRF</th>
<th align="center">Complexity</th>
<th align="center">Flag colors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">&lt; 0.5</td>
<td align="center">&lt; 1</td>
<td align="center">Severe</td>
<td align="center">&lt; 0.5</td>
<td align="center">Concerning</td>
<td align="center">Orange</td>
</tr>
<tr class="even">
<td align="center">0.5 ≤ PBC1 &lt; 0.8</td>
<td align="center">1 ≤ PBC2 &lt; 3</td>
<td align="center">Moderate</td>
<td align="center">0.5 ≤ NRF &lt; 0.8</td>
<td align="center">Acceptable</td>
<td align="center">Yellow</td>
</tr>
<tr class="odd">
<td align="center">0.8 ≤ PBC1 &lt; 0.9</td>
<td align="center">3 ≤ PBC2 &lt; 10</td>
<td align="center">Mild</td>
<td align="center">0.8 ≤ NRF &lt; 0.9</td>
<td align="center">Compliant</td>
<td align="center">None</td>
</tr>
<tr class="even">
<td align="center">≥ 0.9</td>
<td align="center">≥ 10</td>
<td align="center">None</td>
<td align="center">&gt; 0.9</td>
<td align="center">Ideal</td>
<td align="center">None</td>
</tr>
</tbody>
</table>
<p>See full snakemake script for details: code/dunnart_peak_calling/dunnart_snakefile.</p>
<p>Table. Read alignment quality metrics before and aftering filtering steps | ** ** <br> ** ** | ** ** | ** <strong>read counts</strong> ** | ** ** | ** ** | ** ** | ** ** | ** ** | | :————————–: | :—————–: | :————————–: | :———————-: | :————————: | :————–: | :—————: | :—————-: | | <strong>Sample</strong> | <strong>Antibody</strong> | <em>bowtie2 alignment</em> | <em>filtered reads*</em> | <em>remove duplicates</em> | <strong>NRF*</strong> | <strong>PBC1*</strong> | <strong>PBC2*</strong> | | A-1 | input | 130587134 | 87165286 | 47645260 | 07773602 | 07774519 | 4445728 | | A-2 | H3K4me3 | 103324742 | 62678360 | 35617708 | 07789709 | 07789661 | 47754989 | | A-3 | H3K27ac | 131071676 | 86503078 | 45700530 | 07761768 | 07761687 | 41196586 | | B-1 | input | 111574640 | 72303316 | 44433864 | 08820313 | 08819974 | 55545324 | | B-2 | H3K4me3 | 114146802 | 67048994 | 42357180 | 08831032 | 08830749 | 58898224 | | B-3 | H3K27ac | 104714846 | 66615704 | 43939460 | 08847255 | 08847022 | 65529168 |</p>
<p>* calculated from a subsample of ~10-15M aligned reads prior to removing duplicates</p>
<p>^ excluded low quality (MAPQ 30) and orphaned reads</p>
</div>
<div id="read-qc-with-deeptools" class="section level2">
<h2>Read QC with deepTools</h2>
<div id="plot-correlation-between-bam-files" class="section level3">
<h3>Plot correlation between BAM files</h3>
<p><code>plotCorrelation</code> computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions.</p>
<p>This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.</p>
<div class="figure">
<embed src="pearsoncor_multibamsum.pdf" />
<p class="caption">Pearson correlation dunnart BAM files</p>
</div>
</div>
<div id="fingerprint-plots" class="section level3">
<h3>Fingerprint plots</h3>
<p>This tool samples indexed BAM files and plots a profile of cumulative read coverages for each. All reads overlapping a window (bin) of the specified length are counted; these counts are sorted and the cumulative sum is finally plotted.</p>
<p>It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample.</p>
<p>An ideal input with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin etc.) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.</p>
<div class="figure">
<embed src="multiBAM_fingerprint.pdf" />
<p class="caption">Fingerprint plot for dunnart BAM files</p>
</div>
</div>
<div id="bam-pe-fragment-size" class="section level3">
<h3>BAM PE fragment size</h3>
<p>This tool calculates the fragment sizes for read pairs given a BAM file from paired-end sequencing. Several regions are sampled depending on the size of the genome and number of processors to estimate thesummary statistics on the fragment lengths.</p>
<div class="figure">
<embed src="bamPEFragmentSize_hist.pdf" />
<p class="caption">Fragment size for read pairs in a BAM file</p>
</div>
</div>
</div>
<div id="cross-correlation-analysis" class="section level2">
<h2>Cross-correlation analysis</h2>
<p>Information from: <a href="https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit" class="uri">https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit</a></p>
<p>This set of programs operate on mapped Illumina single-end read datasets in tagAlign or BAM format. Because my data is paired-end I need to only use the forward read.</p>
<p>A high-quality ChIP-seq experiment will produce significant clustering of enriched DNA sequence tags/reads at locations bound by the protein of interest; the expectation is that we can observe a bimodal enrichment of reads (sequence tags) on both the forward and the reverse strands.</p>
<p>Cross-correlation analysis is done on a filtered (but not-deduped) and subsampled BAM. There is a special fastq trimming for cross-correlation analysis. Read1 fastq is trimmed to 50bp first using trimfastq.py (last modified 2017/11/08, <a href="https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py" class="uri">https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py</a>). And then it is separately mapped as SE. Reads are filtered but duplicates are not removed. Then 15 million reads are randomly sampled and used for cross-correlation analysis.</p>
<p>ENCODE standards:</p>
<table>
<colgroup>
<col width="4%" />
<col width="14%" />
<col width="80%" />
</colgroup>
<thead>
<tr class="header">
<th>col.</th>
<th>abbreviation</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Filename</td>
<td>tagAlign/BAM filename</td>
</tr>
<tr class="even">
<td>2</td>
<td>numReads</td>
<td>effective sequencing depth i.e. total number of mapped reads in input file</td>
</tr>
<tr class="odd">
<td>3</td>
<td>estFragLen</td>
<td>comma separated strand cross-correlation peak(s) in decreasing order of correlation.</td>
</tr>
<tr class="even">
<td>4</td>
<td>corr_estFragLen</td>
<td>comma separated strand cross-correlation value(s) in decreasing order (COL2 follows the same order)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>phantomPeak</td>
<td>Read length/phantom peak strand shift</td>
</tr>
<tr class="even">
<td>6</td>
<td>corr_phantomPeak</td>
<td>Correlation value at phantom peak</td>
</tr>
<tr class="odd">
<td>7</td>
<td>argmin_corr</td>
<td>strand shift at which cross-correlation is lowest</td>
</tr>
<tr class="even">
<td>8</td>
<td>min_corr</td>
<td>minimum value of cross-correlation</td>
</tr>
<tr class="odd">
<td>9</td>
<td>NSC</td>
<td>Normalized strand cross-correlation coefficient (NSC) = COL4 / COL8</td>
</tr>
<tr class="even">
<td>10</td>
<td>RSC</td>
<td>Relative strand cross-correlation coefficient (RSC) = (COL4 - COL8) / (COL6 - COL8)</td>
</tr>
<tr class="odd">
<td>11</td>
<td>QualityTag</td>
<td>Quality tag based on thresholded RSC (codes= -2:veryLow, -1:Low, 0:Medium, 1:High, 2:veryHigh)</td>
</tr>
</tbody>
</table>
<p>NSC; NSC&gt;1.1 (higher values indicate more enrichment; 1 = no enrichment)</p>
<p>RSC; RSC&gt;0.8 (0 = no signal; &lt;1 low quality ChIP; &gt;1 high enrichment</p>
<p>Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High; 2:veryHigh)</p>
<pre class="r"><code>library(data.table)
library(knitr)
A1 = fread(&quot;output/qc/A-1_input_filt_15Mreads.SE.cc.qc&quot;)
colnames(A1) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)
A2 = fread(&quot;output/qc/A-2_H3K4me3_filt_15Mreads.SE.cc.qc&quot;)
colnames(A2) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)
A3 = fread(&quot;output/qc/A-3_H3K27ac_filt_15Mreads.SE.cc.qc&quot;)
colnames(A3) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)
B1 = fread(&quot;output/qc/B-1_input_filt_15Mreads.SE.cc.qc&quot;)
colnames(B1) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)
B2 = fread(&quot;output/qc/B-2_H3K4me3_filt_15Mreads.SE.cc.qc&quot;)
colnames(B2) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)
B3 = fread(&quot;output/qc/B-3_H3K27ac_filt_15Mreads.SE.cc.qc&quot;)
colnames(B3) = c(&quot;Filename&quot;, &quot;numReads&quot;, &quot;estFragLen&quot;, &quot;corr_estFragLen&quot;, &quot;PhantomPeak&quot;, &quot;corr_phantomPeak&quot;, &quot;argmin_corr&quot;, &quot;min_corr&quot;, &quot;phantomPeakCoef&quot;, &quot;relPhantomPeakCoef&quot;, &quot;QualityTag&quot;)</code></pre>
<table>
<caption>A1 H3K27ac cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A-1_input.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">95,135,150</td>
<td align="left">0.123657819405274,0.123466870849122,0.123426393106245</td>
<td align="right">50</td>
<td align="right">0.125339</td>
<td align="right">1500</td>
<td align="right">0.1220062</td>
<td align="right">1.013537</td>
<td align="right">0.495558</td>
<td align="right">-1</td>
</tr>
</tbody>
</table>
<table>
<caption>A2 H3K4me3 cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A-2_H3K4me3.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">205,225,240</td>
<td align="left">0.264180702500576,0.264046178090335,0.263856306475591</td>
<td align="right">50</td>
<td align="right">0.2533387</td>
<td align="right">1500</td>
<td align="right">0.1496324</td>
<td align="right">1.765532</td>
<td align="right">1.104545</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<table>
<caption>A3 input control cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A-3_H3K27ac.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">210,220,235</td>
<td align="left">0.22968659727008,0.229482796607544,0.229305284391192</td>
<td align="right">55</td>
<td align="right">0.2184643</td>
<td align="right">1500</td>
<td align="right">0.1570855</td>
<td align="right">1.462175</td>
<td align="right">1.182836</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<table>
<caption>B1 H3K27ac control cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B-1_input.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">115,130,150</td>
<td align="left">0.128496809011662,0.128442767656697,0.12834049936991</td>
<td align="right">50</td>
<td align="right">0.1304347</td>
<td align="right">1500</td>
<td align="right">0.126898</td>
<td align="right">1.012599</td>
<td align="right">0.4520538</td>
<td align="right">-1</td>
</tr>
</tbody>
</table>
<table>
<caption>B2 H3K4me3 control cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B-2_H3K4me3.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">210,225,245</td>
<td align="left">0.316307968017767,0.316149524309277,0.315180730244617</td>
<td align="right">50</td>
<td align="right">0.3014805</td>
<td align="right">1500</td>
<td align="right">0.1675044</td>
<td align="right">1.888356</td>
<td align="right">1.110672</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<table>
<caption>B3 input control cross-correlation analysis</caption>
<colgroup>
<col width="21%" />
<col width="4%" />
<col width="5%" />
<col width="24%" />
<col width="5%" />
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Filename</th>
<th align="right">numReads</th>
<th align="left">estFragLen</th>
<th align="left">corr_estFragLen</th>
<th align="right">PhantomPeak</th>
<th align="right">corr_phantomPeak</th>
<th align="right">argmin_corr</th>
<th align="right">min_corr</th>
<th align="right">phantomPeakCoef</th>
<th align="right">relPhantomPeakCoef</th>
<th align="right">QualityTag</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B-3_H3K27ac.filt.sample.15Mreads.SE.tagAlign.gz</td>
<td align="right">1.5e+07</td>
<td align="left">220,230,255</td>
<td align="left">0.229222829886832,0.228995326028105,0.228262435658256</td>
<td align="right">55</td>
<td align="right">0.2203839</td>
<td align="right">1500</td>
<td align="right">0.1693505</td>
<td align="right">1.353541</td>
<td align="right">1.173198</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div id="peak-qc" class="section level2">
<h2>Peak QC</h2>
<div id="number-of-peaks" class="section level3">
<h3>Number of peaks</h3>
<pre class="bash"><code>less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l</code></pre>
<pre><code>171653</code></pre>
<pre class="bash"><code>less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l</code></pre>
<pre><code>118626</code></pre>
<pre class="bash"><code>less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l</code></pre>
<pre><code>173791</code></pre>
<pre class="bash"><code>less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l</code></pre>
<pre><code>130443</code></pre>
</div>
<div id="fraction-of-reads-in-peaks" class="section level3">
<h3>Fraction of reads in peaks</h3>
<p>A-2_H3K4me3 fraction of reads in peaks</p>
<pre class="bash"><code>less output/qc/A-2_H3K4me3_vs_A-1_input.frip_default.txt</code></pre>
<pre><code>Number of reads that intersect with peaks =  10471951
Number of total reads =  17808854
Fraction of Reads in Peak =  0.5880193638512619</code></pre>
<p>A3 H3K27ac fraction of reads in peaks</p>
<pre class="bash"><code>less output/qc/A-3_H3K27ac_vs_A-1_input.frip_default.txt</code></pre>
<pre><code>Number of reads that intersect with peaks =  14183696
Number of total reads =  22850265
Fraction of Reads in Peak =  0.6207234795745257</code></pre>
<p>B2 H3K4me3 fraction of reads in peaks</p>
<pre class="bash"><code>less output/qc/B-2_H3K4me3_vs_B-1_input.frip_default.txt</code></pre>
<pre><code>Number of reads that intersect with peaks =  13916104
Number of total reads =  21178590
Fraction of Reads in Peak =  0.6570835924393456</code></pre>
<p>B3 H3K27ac fraction of reads in peaks</p>
<pre class="bash"><code>less output/qc/B-3_H3K27ac_vs_B-1_input.frip_default.txt</code></pre>
<pre><code>Number of reads that intersect with peaks =  13467294
Number of total reads =  21969730
Fraction of Reads in Peak =  0.6129931501206433</code></pre>
</div>
<div id="consensus-peaksets-for-biological-replicates" class="section level3">
<h3>Consensus peaksets for biological replicates</h3>
<p>Number of consensus peaks</p>
<pre class="bash"><code>less output/peaks/H3K4me3_overlap_default.narrowPeak | wc -l</code></pre>
<pre><code>80989</code></pre>
<pre class="bash"><code>less output/peaks/H3K27ac_overlap_default.narrowPeak | wc -l</code></pre>
<pre><code>118528</code></pre>
</div>
</div>
</div>
<div id="group-peaks-into-promoter--and-enhancer--associated-peaks" class="section level1">
<h1>Group peaks into promoter- and enhancer- associated peaks</h1>
<p>Find uniquely H3K4me3 sites (i.e. peaks that don’t overlap with H3K27ac):</p>
<pre class="bash"><code>module load bedtools
bedtools intersect -v -a output/peaks/H3K4me3_overlap_default.narrowPeak -b output/peaks/H3K27ac_overlap_default.narrowPeak &gt; output/peaks/H3K4me3_only.narrowPeak
less output/peaks/H3K4me3_only.narrowPeak | wc -l</code></pre>
<pre><code>bash: module: command not found
bash: line 1: bedtools: command not found
0</code></pre>
<p>Find unique H3K27ac sites (i.e. peaks that don’t overlap with H3K4me3):</p>
<pre class="bash"><code>module load bedtools
bedtools intersect -v -a output/peaks/H3K27ac_overlap_default.narrowPeak -b output/peaks/H3K4me3_overlap_default.narrowPeak &gt; output/peaks/H3K27ac_only.narrowPeak
less output/peaks/H3K27ac_only.narrowPeak | wc -l</code></pre>
<pre><code>bash: module: command not found
bash: line 1: bedtools: command not found
0</code></pre>
<p>Find peaks common between H3K27ac &amp; H3K4me3 with a reciprocal overlap of at least 50%.</p>
<pre class="bash"><code>module load bedtools
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_overlap_default.narrowPeak -b output/peaks/H3K27ac_overlap_default.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac.narrowPeak
less output/peaks/H3K4me3_and_H3K27ac.narrowPeak | wc -l</code></pre>
<pre><code>bash: module: command not found
bash: line 1: bedtools: command not found
0</code></pre>
<p>Combine files</p>
<pre class="bash"><code>cat output/peaks/H3K4me3_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac.narrowPeak | sort -k 4,4 &gt; output/peaks/dunnart_promoter_peaks.narrowPeak
cat output/peaks/H3K27ac_only.narrowPeak | sort -k 4,4 &gt; output/peaks/dunnart_enhancer_peaks.narrowPeak
less output/peaks/dunnart_promoter_peaks.narrowPeak | wc -l
less output/peaks/dunnart_enhancer_peaks.narrowPeak | wc -l</code></pre>
<pre><code>0
0</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS/LAPACK: /usr/local/easybuild-2019/easybuild/software/compiler/gcc/10.2.0/openblas/0.3.12/lib/libopenblas_haswellp-r0.3.12.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.33        data.table_1.14.0 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        highr_0.9         jquerylib_0.1.4   pillar_1.6.1     
 [5] compiler_4.1.0    bslib_0.2.5.1     later_1.2.0       git2r_0.28.0     
 [9] tools_4.1.0       getPass_0.2-2     digest_0.6.27     jsonlite_1.7.2   
[13] evaluate_0.14     lifecycle_1.0.1   tibble_3.1.2      pkgconfig_2.0.3  
[17] rlang_1.0.2       cli_2.5.0         rstudioapi_0.13   yaml_2.2.1       
[21] xfun_0.23         httr_1.4.2        stringr_1.4.0     sass_0.4.0       
[25] fs_1.5.0          vctrs_0.3.8       rprojroot_2.0.2   glue_1.4.2       
[29] R6_2.5.0          processx_3.5.2    fansi_0.5.0       rmarkdown_2.8    
[33] callr_3.7.0       magrittr_2.0.1    whisker_0.4       ps_1.6.0         
[37] promises_1.2.0.1  ellipsis_0.3.2    htmltools_0.5.1.1 httpuv_1.6.1     
[41] utf8_1.2.1        stringi_1.6.2     crayon_1.4.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
