<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">chipseq_cross_species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lecook/chipseq-cross-species">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chipseq-cross-species/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestree91574f92ce5bfe0eee9a26f4f98ddb3fba8d643etargetblank91574f9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lecook/chipseq-cross-species/tree/91574f92ce5bfe0eee9a26f4f98ddb3fba8d643e" target="_blank">91574f9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestree91574f92ce5bfe0eee9a26f4f98ddb3fba8d643etargetblank91574f9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lecook/chipseq-cross-species/tree/91574f92ce5bfe0eee9a26f4f98ddb3fba8d643e" target="_blank">91574f9</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    data/genomic_data/
    Ignored:    output/annotations/
    Ignored:    output/bam_files/
    Ignored:    output/filtered_peaks/
    Ignored:    output/logs/
    Ignored:    output/peaks/
    Ignored:    output/plots/
    Ignored:    results/

Untracked files:
    Untracked:  .snakemake/
    Untracked:  H3K27ac_only.narrowPeak
    Untracked:  H3K4me3_and_H3K27ac.narrowPeak
    Untracked:  H3K4me3_only.narrowPeak
    Untracked:  code/basic_wrapper.slurm
    Untracked:  code/configs/metadata.tsv
    Untracked:  code/slurmjob.stderr
    Untracked:  code/slurmjob.stdout
    Untracked:  code/trimfastq.py
    Untracked:  data/raw_reads/
    Untracked:  dunnart_enhancer_peaks.narrowPeak
    Untracked:  dunnart_promoter_peaks.narrowPeak
    Untracked:  output/qc/
    Untracked:  output/rnaseq/
    Untracked:  slurm-33634648.out
    Untracked:  slurm-33634649.out
    Untracked:  slurm-33634650.out
    Untracked:  slurm-33634651.out
    Untracked:  slurm-33634652.out
    Untracked:  slurm-33634653.out
    Untracked:  slurm-33634669.out
    Untracked:  slurm-33634682.out
    Untracked:  slurm-33638735.out
    Untracked:  slurm-33638749.out
    Untracked:  slurm-33638750.out
    Untracked:  slurm-33638751.out
    Untracked:  slurm-33639342.out
    Untracked:  slurm-33639343.out
    Untracked:  slurm-33639344.out
    Untracked:  slurm-33639345.out
    Untracked:  slurm-33639346.out
    Untracked:  slurm-33639347.out
    Untracked:  slurm-33652974.out
    Untracked:  slurm-33652975.out
    Untracked:  slurm-33652976.out
    Untracked:  slurm-33652977.out
    Untracked:  slurm-33652978.out
    Untracked:  slurm-33652979.out
    Untracked:  slurm-33653279.out
    Untracked:  slurm-33653280.out
    Untracked:  slurm-33653281.out
    Untracked:  slurm-33653282.out
    Untracked:  slurm-33653283.out
    Untracked:  slurm-33653284.out
    Untracked:  slurm-33654225.out
    Untracked:  slurm-33654226.out
    Untracked:  slurm-33654227.out
    Untracked:  slurm-33654228.out
    Untracked:  slurm-33654229.out
    Untracked:  slurm-33654230.out
    Untracked:  slurm-33654264.out
    Untracked:  slurm-33654265.out
    Untracked:  slurm-33654266.out
    Untracked:  slurm-33654267.out
    Untracked:  slurm-33655778.out
    Untracked:  slurm-33655779.out
    Untracked:  slurm-33655780.out
    Untracked:  slurm-33655781.out
    Untracked:  slurm-33655782.out
    Untracked:  slurm-33655783.out
    Untracked:  slurm-33655867.out
    Untracked:  slurm-33655869.out
    Untracked:  slurm-33655871.out
    Untracked:  slurm-33655873.out
    Untracked:  slurm-33655936.out
    Untracked:  slurm-33655937.out
    Untracked:  slurm-33655964.out
    Untracked:  slurm-33655966.out
    Untracked:  slurm-33656003.out
    Untracked:  slurm-33656023.out
    Untracked:  slurm-33656024.out
    Untracked:  slurm-33656025.out
    Untracked:  slurm-33656027.out
    Untracked:  slurm-33656056.out
    Untracked:  slurm-33656089.out
    Untracked:  slurm-33656090.out
    Untracked:  slurm-33656102.out
    Untracked:  slurm-33656104.out
    Untracked:  slurm-33656170.out
    Untracked:  slurm-33656224.out
    Untracked:  slurm-33656227.out
    Untracked:  slurm-33656228.out
    Untracked:  slurm-33656312.out
    Untracked:  slurm-33656357.out
    Untracked:  slurm-33656358.out
    Untracked:  slurm-33656364.out
    Untracked:  slurm-33656365.out
    Untracked:  slurm-33656409.out
    Untracked:  slurm-33656545.out
    Untracked:  slurm-33657362.out
    Untracked:  slurm-33657363.out
    Untracked:  slurm-33657364.out
    Untracked:  slurm-33657365.out
    Untracked:  slurm-33657366.out
    Untracked:  slurm-33657367.out
    Untracked:  slurm-33657375.out
    Untracked:  slurm-33657376.out
    Untracked:  slurm-33657625.out
    Untracked:  slurm-33657626.out
    Untracked:  slurm-33657627.out
    Untracked:  slurm-33657628.out
    Untracked:  slurm-33657629.out
    Untracked:  slurm-33657630.out
    Untracked:  slurm-33657636.out
    Untracked:  slurm-33657638.out
    Untracked:  slurm-33657645.out
    Untracked:  slurm-33657646.out
    Untracked:  slurm-33657677.out
    Untracked:  slurm-33657688.out

Unstaged changes:
    Deleted:    code/dunnart_peak_calling/trimfastq.py

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/a97d3c57c638c074a308452653233ad3bfc59f48/analysis/index.Rmd" target="_blank">a97d3c5</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/67e8b13e9bc9011aa8daf2e473ff5747a7cdfd62/docs/index.html" target="_blank">67e8b13</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/9d9a99533caf5d7803ed5576afd8786ef296a7ea/analysis/index.Rmd" target="_blank">9d9a995</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
udpated
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/275d2299013a6db02d57b94c22d287788ec16d15/docs/index.html" target="_blank">275d229</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5fb3661d2eb6bfa7968354bd435070ebf80e7a21/analysis/index.Rmd" target="_blank">5fb3661</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
clean up
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/c2eeed7fdd41366e809bacad538f2e5afed933cf/analysis/index.Rmd" target="_blank">c2eeed7</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
debugged
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/f331cc03bdde66533bc2d74a84176e40cec8427d/analysis/index.Rmd" target="_blank">f331cc0</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/6134b21a4152be1e3538d732c95044d5186d6065/docs/index.html" target="_blank">6134b21</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/6fa3b022b953ab06c2f627f8e707dc6bc55cb284/analysis/index.Rmd" target="_blank">6fa3b02</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
update website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/7e126783b2a12cd7d7c9dd4e972a3492a95e92d6/docs/index.html" target="_blank">7e12678</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/bc0b45f8c7e595bf8435bfb322ccaaa07d6b929b/analysis/index.Rmd" target="_blank">bc0b45f</a>
</td>
<td>
GitHub
</td>
<td>
2022-02-28
</td>
<td>
Update index.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/277f9e450a12bc300ab2677db8b2640355fb7c53/docs/index.html" target="_blank">277f9e4</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/946cd8b86f9c09fa8b4c44c30d452129e61cde37/analysis/index.Rmd" target="_blank">946cd8b</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/072d4e4a258999fb4b217a6e9b063ed950504bf3/docs/index.html" target="_blank">072d4e4</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/93956a3e0419745df2a204380e1ac8d26b1eca4f/analysis/index.Rmd" target="_blank">93956a3</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
added html link for dunnart data pre-processing file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/fc2805319d09230ecf9d007aa69d51fc34069b27/docs/index.html" target="_blank">fc28053</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/28e36bd76ca97d1fae0d2d00e12cfb2ef01b80d9/docs/index.html" target="_blank">28e36bd</a>
</td>
<td>
lecook
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5eaf07ccf2bf0631e9683e75665c016f244ff6d1/analysis/index.Rmd" target="_blank">5eaf07c</a>
</td>
<td>
lecook
</td>
<td>
2022-02-23
</td>
<td>
add website links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/dd843bfd49e430b4e965c4fbf4593cad03afb0fa/docs/index.html" target="_blank">dd843bf</a>
</td>
<td>
lecook
</td>
<td>
2022-02-23
</td>
<td>
make website links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/a72b54032e2f97b05fdb2e1c428e8e95c93b31e3/docs/index.html" target="_blank">a72b540</a>
</td>
<td>
lecook
</td>
<td>
2022-02-22
</td>
<td>
add html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/7c723e062cddd499cd3f643eca02814666773ad7/analysis/index.Rmd" target="_blank">7c723e0</a>
</td>
<td>
lecook
</td>
<td>
2022-02-09
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Laura Cook</p>
<div id="workflow-and-analysis-of-comparative-craniofacial-chip-seq-work-in-the-fat-tailed-dunnart-and-mouse" class="section level1">
<h1>Workflow and analysis of comparative craniofacial ChIP-seq work in the fat-tailed dunnart and mouse</h1>
<p>This project involves a number of analysis for the generation of ChIP-seq data for the fat-tailed dunnart for comparison with the exisiting mouse data in ENCODE.</p>
<div id="data" class="section level2">
<h2>Data</h2>
<ul>
<li><p><a href="dunnart_experiment_overview.html">Dunnart experimental design and data generation</a></p></li>
<li><p><a href="mouse_data_ENCODE.html">Mouse data from ENCODE</a></p></li>
</ul>
</div>
<div id="data-preparation-and-peak-calling" class="section level2">
<h2>Data preparation and peak calling</h2>
<ul>
<li><p><a href="dunnart_data_processing.html">Dunnart data pre-processing</a></p>
<ul>
<li>Outlines read mapping, filtering, QC and peak calling.</li>
</ul></li>
<li><p><a href="mouse_dunnart_data_processing_for_comparison.html">Normalised mouse and dunnart read pre-processing</a></p>
<ul>
<li>Outlines filtering aligned reads from ENCODE, QC and peak calling.</li>
</ul></li>
</ul>
</div>
<div id="data-analyses" class="section level2">
<h2>Data analyses</h2>
<ul>
<li><p><a href="whole_genome_alignment.html">Whole genome alignment between mouse and dunnart</a></p></li>
<li><p><a href="dunnart_peak_characterisation.html">Dunnart peak characterisation</a></p></li>
<li><p><a href="cross-species-peak-comparison.html">Cross-species peak comparisons</a></p>
<ul>
<li><p><a href="peak_level_comparisons.html">Peak level</a></p></li>
<li><p><a href="gene_level_comparisons.html">Gene level</a></p></li>
<li><p><a href="tcseq_expression_analysis.html">Incorporating temporal mouse gene expression data</a></p></li>
</ul></li>
</ul>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
