<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="lecook" />

<meta name="date" content="2022-02-23" />

<title>Processing mouse and dunnart data for comparison</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">chipseq_cross_species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/lecook/chipseq-cross-species">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Processing mouse and dunnart data for comparison</h1>
<h4 class="author">lecook</h4>
<h4 class="date">2022-02-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chipseq-cross-species/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220209code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220209)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220209code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220209)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestreeeca7e889e5c91a0891f00cc246ce3c3bd0a2c9c7targetblankeca7e88a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/lecook/chipseq-cross-species/tree/eca7e889e5c91a0891f00cc246ce3c3bd0a2c9c7" target="_blank">eca7e88</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomlecookchipseqcrossspeciestreeeca7e889e5c91a0891f00cc246ce3c3bd0a2c9c7targetblankeca7e88a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/lecook/chipseq-cross-species/tree/eca7e889e5c91a0891f00cc246ce3c3bd0a2c9c7" target="_blank">eca7e88</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    data/genomic_data/
    Ignored:    output/annotations/
    Ignored:    output/bam_files/
    Ignored:    output/filtered_peaks/
    Ignored:    output/logs/
    Ignored:    output/peaks/
    Ignored:    output/plots/
    Ignored:    output/qc/A-1_input.SeqDepthNorm.bw
    Ignored:    output/qc/A-1_input.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-1_input.ccurve.txt
    Ignored:    output/qc/A-1_input.extrap.txt
    Ignored:    output/qc/A-1_input_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-1_input_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-2_H3K4me3.SeqDepthNorm.bw
    Ignored:    output/qc/A-2_H3K4me3.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-2_H3K4me3.ccurve.txt
    Ignored:    output/qc/A-2_H3K4me3.extrap.txt
    Ignored:    output/qc/A-2_H3K4me3_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-2_H3K4me3_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_A-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/A-2_H3K4me3_vs_B-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/A-3_H3K27ac.SeqDepthNorm.bw
    Ignored:    output/qc/A-3_H3K27ac.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/A-3_H3K27ac.ccurve.txt
    Ignored:    output/qc/A-3_H3K27ac.extrap.txt
    Ignored:    output/qc/A-3_H3K27ac_R1_trimmed.fastq.gz
    Ignored:    output/qc/A-3_H3K27ac_est_lib_complex_metrics.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_A-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/A-3_H3K27ac_vs_B-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/B-1_input.SeqDepthNorm.bw
    Ignored:    output/qc/B-1_input.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-1_input.ccurve.txt
    Ignored:    output/qc/B-1_input.extrap.txt
    Ignored:    output/qc/B-1_input_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-1_input_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-2_H3K4me3.SeqDepthNorm.bw
    Ignored:    output/qc/B-2_H3K4me3.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-2_H3K4me3.ccurve.txt
    Ignored:    output/qc/B-2_H3K4me3.extrap.txt
    Ignored:    output/qc/B-2_H3K4me3_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-2_H3K4me3_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_A-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/B-2_H3K4me3_vs_B-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/B-3_H3K27ac.SeqDepthNorm.bw
    Ignored:    output/qc/B-3_H3K27ac.SeqDepthNorm_dunnart_downSampled.bw
    Ignored:    output/qc/B-3_H3K27ac.ccurve.txt
    Ignored:    output/qc/B-3_H3K27ac.extrap.txt
    Ignored:    output/qc/B-3_H3K27ac_R1_trimmed.fastq.gz
    Ignored:    output/qc/B-3_H3K27ac_est_lib_complex_metrics.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_A-1_input.frip_default.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_A-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_B-1_input.frip_default.txt
    Ignored:    output/qc/B-3_H3K27ac_vs_B-1_input.frip_default_dunnart_downSampled.txt
    Ignored:    output/qc/ENCFF665QBJ_E14.5_H3K4me3.SeqDepthNorm_mouse.bw
    Ignored:    output/qc/H3K27ac_multiBAM_fingerprint_metrics.txt
    Ignored:    output/qc/H3K27ac_multiBAM_fingerprint_rawcounts.txt
    Ignored:    output/qc/H3K27ac_multibamsum.npz
    Ignored:    output/qc/H3K27ac_multibamsum.tab
    Ignored:    output/qc/H3K27ac_pearsoncor_multibamsum_matrix.txt
    Ignored:    output/qc/H3K27ac_plot_coverage_rawcounts.tab
    Ignored:    output/qc/bamPEFragmentSize_rawcounts.tab
    Ignored:    output/qc/bamPEFragmentSize_rawcounts_dunnart_downSampled.tab
    Ignored:    output/qc/multiBAM_fingerprint_metrics.txt
    Ignored:    output/qc/multiBAM_fingerprint_metrics_dunnart_downSampled.txt
    Ignored:    output/qc/multiBAM_fingerprint_rawcounts.txt
    Ignored:    output/qc/multiBAM_fingerprint_rawcounts_dunnart_downSampled.txt
    Ignored:    output/qc/multibamsum.npz
    Ignored:    output/qc/multibamsum.tab
    Ignored:    output/qc/multibamsum_dunnart_downSampled.npz
    Ignored:    output/qc/multibamsum_dunnart_downSampled.tab
    Ignored:    output/qc/pearsoncor_multibamsum_matrix.txt
    Ignored:    output/qc/pearsoncor_multibamsum_matrix_dunnart_downSampled.txt
    Ignored:    output/wga/

Untracked files:
    Untracked:  .snakemake/
    Untracked:  Rplots.pdf
    Untracked:  code/.swp
    Untracked:  counts.out
    Untracked:  data/raw_reads/
    Untracked:  names.out
    Untracked:  output/qc/ENCFF002HZV_E14.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF002HZV_E14.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF011NFM_E12.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF011NFM_E12.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF045IPK_E10.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF058AUT_E12.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF058AUT_E12.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF117QRC_E13.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF117QRC_E13.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF124TAB_E13.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF124UYX_E10.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF157KEH_E10.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF157KEH_E10.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF182XFG_E15.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF182XFG_E15.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF182ZPF_E12.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF182ZPF_E12.5_H3K4me3.pbc.qc
    Untracked:  output/qc/ENCFF184CUE_E11.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF184CUE_E11.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF194ORC_E13.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF194ORC_E13.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF194ORC_E13.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF194ORC_E13.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF203JQV_E12.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF203JQV_E12.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF213EBC_E10.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF248PGK_E13.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF248PGK_E13.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF258KCR_E15.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF290ZNF_E13.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF327VAO_E14.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF376FGM_E11.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF376FGM_E11.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF394TZN_E12.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF394TZN_E12.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF401BKM_E15.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF401BKM_E15.5_H3K4me3.pbc.qc
    Untracked:  output/qc/ENCFF485UDC_E13.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF512SFE_E11.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF512SFE_E11.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF515PKL_E11.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF515PKL_E11.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF515PKL_E11.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF515PKL_E11.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF548BRR_E10.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac_filt_15Mreads.SE.cc.plot.pdf
    Untracked:  output/qc/ENCFF584JFB_E15.5_H3K27ac_filt_15Mreads.SE.cc.qc
    Untracked:  output/qc/ENCFF665QBJ_E14.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF707WKL_E15.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF717QDV_E11.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF724DMU_E14.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF727QTS_E15.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF727QTS_E15.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF760QYZ_E11.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF784ORI_E14.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF784ORI_E14.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF825AVI_E10.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF825AVI_E10.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.dedup.flagstat.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.downSampled.flagstat.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.dupmark.flagstat.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.dupmark.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.pbc.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.q30.flagstat.qc
    Untracked:  output/qc/ENCFF902HAR_E14.5_H3K27ac.unfiltered.flagstat.qc
    Untracked:  output/qc/ENCFF941QJZ_E12.5_H3K4me3.dupmark.qc
    Untracked:  output/qc/H3K27ac_multiBAM_fingerprint.pdf
    Untracked:  output/qc/H3K27ac_overlap_default_dunnart_downSampled.frip
    Untracked:  output/qc/H3K27ac_pearsoncor_multibamsum.pdf
    Untracked:  output/qc/H3K27ac_plot_coverage.pdf
    Untracked:  output/qc/H3K4me3_overlap_default_dunnart_downSampled.frip
    Untracked:  output/qc/ucsc_alignment/
    Untracked:  output/rnaseq/
    Untracked:  slurm-34005063.out
    Untracked:  slurm-34005064.out
    Untracked:  slurm-34005065.out
    Untracked:  slurm-34005066.out
    Untracked:  slurm-34005067.out
    Untracked:  slurm-34005070.out
    Untracked:  slurm-34005071.out
    Untracked:  slurm-34005072.out
    Untracked:  slurm-34005073.out
    Untracked:  slurm-34005074.out
    Untracked:  slurm-34005075.out
    Untracked:  slurm-34005076.out
    Untracked:  slurm-34005077.out
    Untracked:  slurm-34005078.out
    Untracked:  slurm-34005079.out
    Untracked:  slurm-34005080.out
    Untracked:  slurm-34005082.out
    Untracked:  slurm-34005083.out
    Untracked:  slurm-34005084.out
    Untracked:  slurm-34005085.out
    Untracked:  slurm-34005086.out
    Untracked:  slurm-34005088.out
    Untracked:  slurm-34005089.out
    Untracked:  slurm-34005091.out
    Untracked:  slurm-34005092.out
    Untracked:  slurm-34005095.out
    Untracked:  slurm-34005099.out
    Untracked:  slurm-34005131.out
    Untracked:  slurm-34005133.out
    Untracked:  slurm-34005134.out
    Untracked:  slurm-34005135.out
    Untracked:  slurm-34005137.out
    Untracked:  slurm-34005139.out
    Untracked:  slurm-34005140.out
    Untracked:  slurm-34005148.out
    Untracked:  slurm-34005199.out
    Untracked:  slurm-34005231.out
    Untracked:  slurm-34005236.out
    Untracked:  slurm-34005237.out
    Untracked:  slurm-34005240.out
    Untracked:  slurm-34005241.out
    Untracked:  slurm-34005242.out
    Untracked:  slurm-34005248.out
    Untracked:  slurm-34005252.out
    Untracked:  slurm-34005285.out
    Untracked:  slurm-34005291.out
    Untracked:  slurm-34005298.out
    Untracked:  slurm-34005300.out
    Untracked:  slurm-34005301.out
    Untracked:  slurm-34005861.out
    Untracked:  slurm-34005956.out
    Untracked:  slurm-34005957.out
    Untracked:  slurm-34005958.out
    Untracked:  slurm-34005961.out
    Untracked:  slurm-34005964.out
    Untracked:  slurm-34005966.out
    Untracked:  slurm-34005967.out
    Untracked:  slurm-34006000.out
    Untracked:  slurm-34006004.out
    Untracked:  slurm-34006233.out
    Untracked:  slurm-34006234.out
    Untracked:  slurm-34006235.out
    Untracked:  slurm-34006236.out
    Untracked:  slurm-34006237.out
    Untracked:  slurm-34006238.out
    Untracked:  slurm-34006239.out
    Untracked:  slurm-34006240.out
    Untracked:  slurm-34006241.out
    Untracked:  slurm-34006242.out
    Untracked:  slurm-34006344.out
    Untracked:  slurm-34006345.out
    Untracked:  slurm-34006346.out
    Untracked:  slurm-34006347.out
    Untracked:  slurm-34006348.out
    Untracked:  slurm-34006349.out
    Untracked:  slurm-34006350.out
    Untracked:  slurm-34006382.out
    Untracked:  slurm-34006383.out
    Untracked:  slurm-34030592.out
    Untracked:  slurm-34030593.out
    Untracked:  slurm-34030594.out
    Untracked:  slurm-34030595.out
    Untracked:  slurm-34030596.out
    Untracked:  slurm-34030597.out
    Untracked:  slurm-34030598.out
    Untracked:  slurm-34030605.out
    Untracked:  slurm-34030606.out
    Untracked:  slurm-34055834.out
    Untracked:  slurm-34055835.out
    Untracked:  slurm-34055836.out
    Untracked:  slurm-34055837.out
    Untracked:  slurm-34055838.out
    Untracked:  slurm-34055839.out
    Untracked:  slurm-34055840.out
    Untracked:  slurm-34055841.out
    Untracked:  slurm-34055842.out
    Untracked:  slurm-34055843.out
    Untracked:  slurm-34056100.out
    Untracked:  slurm-34056101.out
    Untracked:  slurm-34056102.out
    Untracked:  slurm-34056103.out
    Untracked:  slurm-34056104.out
    Untracked:  slurm-34056105.out
    Untracked:  slurm-34056115.out

Unstaged changes:
    Modified:   code/configs/cluster.json
    Deleted:    output/qc/H3K4me3_overlap_default.frip

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/mouse_dunnart_data_processing_for_comparison.Rmd</code>) and HTML (<code>docs/mouse_dunnart_data_processing_for_comparison.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/eca7e889e5c91a0891f00cc246ce3c3bd0a2c9c7/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">eca7e88</a>
</td>
<td>
lecook
</td>
<td>
2022-03-17
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/80957c9a90a02cec710ce0f8cd15e8641244f633/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">80957c9</a>
</td>
<td>
lecook
</td>
<td>
2022-03-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/6fc834323e13f3881cab7eefe67574ac2ce79b59/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">6fc8343</a>
</td>
<td>
lecook
</td>
<td>
2022-03-17
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/56ecaa08d5e7587a9c4b024b131c3d42bbac2e34/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">56ecaa0</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/7f26be0ecff584ab45b7620adb28b40d9c879700/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">7f26be0</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/76c6eee54b2884940c1d95b4232f96cb5152685a/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">76c6eee</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/9c12dfe9553d514a18f8018944a42c076dd9d9c3/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">9c12dfe</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/1456cf5d4b4edab011cf243fba7bf63365cdf02b/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">1456cf5</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/9bf4173026bf60fdea7bfbe07c51fe872aa14558/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">9bf4173</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/e662415b33aaf172b2a681e45ec13d6d3888cb69/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">e662415</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/2720905aa083cd4196c50f019f73964a44c438c5/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">2720905</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/7767732b007df2ac1ee9317cd4a4ac30617f78f5/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">7767732</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/c447893f7a66b2d2417b8a7c9ea34fc7fc79a1bc/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">c447893</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/a90d5a07fcb2d1bf2cf001a77bcc1eb97c483331/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">a90d5a0</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/87c414607fc011f7d37319477322a0cf4c6d5fcf/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">87c4146</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/93879c2b9a6074c3400e6ea1d64aa89f079fe6c9/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">93879c2</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/a552f1740dd01542a7c8df053954750b883380af/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">a552f17</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5a17b44c561b7c4ddfb240c23dc74ce6c43c41dd/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">5a17b44</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/11fbc819f26fddfc2c06e289b214ed23ee946415/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">11fbc81</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/dffebdd4f87cdbb71fb2de307adb93737190d592/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">dffebdd</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/4b710689f517a9e0237ca3d49ad2859d9877e4aa/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">4b71068</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/a287d1b19fc6a00cf8b9bc9ce95c5c6ac49d33b8/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">a287d1b</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/eaeea840bb3be227997756b9e02b9e4a52346ef1/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">eaeea84</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/81d567789f5631242ef9c9c8aac6a57877e376a4/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">81d5677</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
fix downsampled
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/c5d4b95bd76eb6f1f06c9f23ed038dfda0097329/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">c5d4b95</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/f7966c8f31274ec644cf22f0577e1de712141852/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">f7966c8</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/040a6a720e003bd1943b9f581512785a84f7e2a1/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">040a6a7</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/f593bedde6d28da650779b4fa4ffe83ebd738464/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">f593bed</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/039df2a3a164e9849009f29ae817b14379d12dd1/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">039df2a</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/9d125bb3fd85409071d27ae50bec786378e772ee/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">9d125bb</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/9bcce4a6d924508c25ec73ebf28a01ba2c2630b3/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">9bcce4a</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/c0a179822002c6a07bccd019633d27cd4ff53b09/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">c0a1798</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/841e0f4a232af03bf28cc6c883b6a3e8b7390df9/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">841e0f4</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/275510f11662865e937b1f347ddbeaa3283117bd/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">275510f</a>
</td>
<td>
lecook
</td>
<td>
2022-03-16
</td>
<td>
wflow_publish(“analysis/mouse_dunnart_data_processing_for_comparison.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/faabd9522c06ff8f990736dbddcc55871fdc363d/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">faabd95</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/2103ae7d50dd723698bf11af8da5e04882dcda03/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">2103ae7</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/323d8b6a03950741027f0b6329a51b8f0c31d796/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">323d8b6</a>
</td>
<td>
lecook
</td>
<td>
2022-03-03
</td>
<td>
copied scripts over
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/ae523af3e11ade3cc2e0c7884c96bfdc5faa66b1/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">ae523af</a>
</td>
<td>
lecook
</td>
<td>
2022-03-02
</td>
<td>
fix code formatting for bam coverage rule
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/a97d3c57c638c074a308452653233ad3bfc59f48/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">a97d3c5</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/lecook/chipseq-cross-species/275d2299013a6db02d57b94c22d287788ec16d15/docs/mouse_dunnart_data_processing_for_comparison.html" target="_blank">275d229</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/784c4ec58b95a43c697b2249147c5a13fa503e82/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">784c4ec</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/bf8fac9ae9ff292e4764f6dd11608faad53c8354/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">bf8fac9</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
updated
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/5fb3661d2eb6bfa7968354bd435070ebf80e7a21/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">5fb3661</a>
</td>
<td>
lecook
</td>
<td>
2022-03-01
</td>
<td>
clean up
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/lecook/chipseq-cross-species/blob/c2eeed7fdd41366e809bacad538f2e5afed933cf/analysis/mouse_dunnart_data_processing_for_comparison.Rmd" target="_blank">c2eeed7</a>
</td>
<td>
lecook
</td>
<td>
2022-02-28
</td>
<td>
debugged
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="pipeline-for-mouse-and-dunnart-peak-calling" class="section level1">
<h1>Pipeline for mouse and dunnart peak calling</h1>
<div id="download-mouse-unfiltered-alignments-from-encode" class="section level2">
<h2>Download mouse unfiltered alignments from ENCODE</h2>
<p>See <a href="mouse_data_ENCODE.html">mouse_data_ENCODE</a> for details on these samples.</p>
<pre class="bash"><code>xargs -L 1 curl -O -J -L &lt; ENCODE_files.txt
xargs -L 1 curl -O -J -L &lt; control_files.txt</code></pre>
</div>
<div id="filter-mouse-alignments" class="section level2">
<h2>Filter mouse alignments</h2>
<p>Uses samtools and picards MarkDuplicates to remove low quality reads. This is based on and in accordance with the ENCODE Guidelines and pipeline (<a href="https://github.com/ENCODE-DCC/chip-seq-pipeline2" class="uri">https://github.com/ENCODE-DCC/chip-seq-pipeline2</a>)</p>
<p><strong>Library Complexity ChIP-seq Standards:</strong></p>
<table>
<thead>
<tr class="header">
<th align="center">PBC1</th>
<th align="center">PBC2</th>
<th align="center">Bottlenecking level</th>
<th align="center">NRF</th>
<th align="center">Complexity</th>
<th align="center">Flag colors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">&lt; 0.5</td>
<td align="center">&lt; 1</td>
<td align="center">Severe</td>
<td align="center">&lt; 0.5</td>
<td align="center">Concerning</td>
<td align="center">Orange</td>
</tr>
<tr class="even">
<td align="center">0.5 ≤ PBC1 &lt; 0.8</td>
<td align="center">1 ≤ PBC2 &lt; 3</td>
<td align="center">Moderate</td>
<td align="center">0.5 ≤ NRF &lt; 0.8</td>
<td align="center">Acceptable</td>
<td align="center">Yellow</td>
</tr>
<tr class="odd">
<td align="center">0.8 ≤ PBC1 &lt; 0.9</td>
<td align="center">3 ≤ PBC2 &lt; 10</td>
<td align="center">Mild</td>
<td align="center">0.8 ≤ NRF &lt; 0.9</td>
<td align="center">Compliant</td>
<td align="center">None</td>
</tr>
<tr class="even">
<td align="center">≥ 0.9</td>
<td align="center">≥ 10</td>
<td align="center">None</td>
<td align="center">&gt; 0.9</td>
<td align="center">Ideal</td>
<td align="center">None</td>
</tr>
</tbody>
</table>
<p>See full snakemake script for details: code/mouse_peak_calling/mouse_H3K4me3 and code/mouse_peak_calling/mouse_H3K27ac</p>
<pre class="r"><code>library(data.table)
library(dplyr)
library(stringr)
library(purrr)
library(tidyverse)
library(knitr)

flagstats &lt;- list.files(&quot;output/qc&quot;, pattern = &quot;*dedup.flagstat.qc|*PPq30.flagstat.qc|*unfiltered.flagstat.qc&quot;, full.names = T) # create list of files in directory
filenames &lt;- sub(&#39;\\.flagstat.qc$&#39;, &#39;&#39;, basename(flagstats)) 
files &lt;- as.list(flagstats)
data &lt;- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = &quot;\t&quot;, quote = &quot;&quot;, na.strings = c(&quot;&quot;, &quot;NA&quot;)))
names(data) &lt;- filenames

stats &lt;- lapply(data, function(x) x[1, ])
stats &lt;- lapply(stats, function(x) data.table(read_count = substr(x, 1, 9)))
stats &lt;- Map(dplyr::mutate, stats, file = names(stats))
stats &lt;- rbindlist(stats,)
names &lt;- str_split_fixed(stats$file, &quot;\\.[A-Za-z]&quot;, 2)
stats2 &lt;- as.data.table(cbind(stats$read_count, names))

colnames(stats2) &lt;- c(&quot;num_reads&quot;, &quot;file&quot;, &quot;processing&quot;)
stats2[stats2$processing == &quot;edup&quot;]$processing &lt;- &quot;duplicates removed&quot; 
stats2[stats2$processing == &quot;nfiltered&quot;]$processing &lt;- &quot;mapped reads&quot; 
stats2[stats2$processing == &quot;Pq30&quot;]$processing &lt;- &quot;remove orphan reads and MAPQ &lt; 30&quot; 

kable(stats2, caption=&quot;Table. Read alignment quality metrics before and after filtering steps&quot;)</code></pre>
<table>
<caption>Table. Read alignment quality metrics before and after filtering steps</caption>
<thead>
<tr class="header">
<th align="left">num_reads</th>
<th align="left">file</th>
<th align="left">processing</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">47645260</td>
<td align="left">A-1_input</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">87165286</td>
<td align="left">A-1_input</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="odd">
<td align="left">130587134</td>
<td align="left">A-1_input</td>
<td align="left">mapped reads</td>
</tr>
<tr class="even">
<td align="left">35617708</td>
<td align="left">A-2_H3K4me3</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="odd">
<td align="left">62678360</td>
<td align="left">A-2_H3K4me3</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="even">
<td align="left">103324742</td>
<td align="left">A-2_H3K4me3</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">45700530</td>
<td align="left">A-3_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">86503078</td>
<td align="left">A-3_H3K27ac</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="odd">
<td align="left">131071676</td>
<td align="left">A-3_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="even">
<td align="left">44433864</td>
<td align="left">B-1_input</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="odd">
<td align="left">72303316</td>
<td align="left">B-1_input</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="even">
<td align="left">111574640</td>
<td align="left">B-1_input</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">42357180</td>
<td align="left">B-2_H3K4me3</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">67048994</td>
<td align="left">B-2_H3K4me3</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="odd">
<td align="left">114146802</td>
<td align="left">B-2_H3K4me3</td>
<td align="left">mapped reads</td>
</tr>
<tr class="even">
<td align="left">43939460</td>
<td align="left">B-3_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="odd">
<td align="left">66615704</td>
<td align="left">B-3_H3K27ac</td>
<td align="left">remove orphan reads and MAPQ &lt; 30</td>
</tr>
<tr class="even">
<td align="left">104714846</td>
<td align="left">B-3_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">61061524</td>
<td align="left">ENCFF213EBC_E10.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">88370370</td>
<td align="left">ENCFF213EBC_E10.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">26624266</td>
<td align="left">ENCFF290ZNF_E13.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">50570025</td>
<td align="left">ENCFF290ZNF_E13.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">19228162</td>
<td align="left">ENCFF327VAO_E14.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">38601448</td>
<td align="left">ENCFF327VAO_E14.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">38069875</td>
<td align="left">ENCFF548BRR_E10.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">50325403</td>
<td align="left">ENCFF548BRR_E10.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">35551034</td>
<td align="left">ENCFF584JFB_E15.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">54217191</td>
<td align="left">ENCFF584JFB_E15.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">27522173</td>
<td align="left">ENCFF707WKL_E15.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">40139064</td>
<td align="left">ENCFF707WKL_E15.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
<tr class="odd">
<td align="left">15203030</td>
<td align="left">ENCFF902HAR_E14.5_H3K27ac</td>
<td align="left">duplicates removed</td>
</tr>
<tr class="even">
<td align="left">28628975</td>
<td align="left">ENCFF902HAR_E14.5_H3K27ac</td>
<td align="left">mapped reads</td>
</tr>
</tbody>
</table>
</div>
<div id="normalise-dunnart-and-mouse-bam-files-to-10m-reads" class="section level2">
<h2>Normalise dunnart and mouse BAM files to 10M reads</h2>
<p>This is so we can compare to mouse peaks.</p>
<p>Based on <a href="https://davemcg.github.io/post/easy-bam-downsampling/" class="uri">https://davemcg.github.io/post/easy-bam-downsampling/</a> script for doing this.</p>
<p>Run script in the directory with the files you want to subsample.</p>
<pre class="bash"><code>cd output/bam_files/
bash subsample.sh</code></pre>
</div>
<div id="read-qc-with-deeptools" class="section level2">
<h2>Read QC with deepTools</h2>
<div id="plot-correlation-between-bam-files" class="section level3">
<h3>Plot correlation between BAM files</h3>
<p><code>plotCorrelation</code> computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions.</p>
<p>This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.</p>
<p><embed src="pearsoncor_multibamsum_dunnart_downSampled.pdf" width="600" height="500" /> <embed src="H3K4me3_pearsoncor_multibamsum_mouse.pdf" width="600" height="500" /> <embed src="H3K27ac_pearsoncor_multibamsum_mouse.pdf" width="600" height="500" /></p>
</div>
<div id="fingerprint-plots" class="section level3">
<h3>Fingerprint plots</h3>
<p>This tool samples indexed BAM files and plots a profile of cumulative read coverages for each. All reads overlapping a window (bin) of the specified length are counted; these counts are sorted and the cumulative sum is finally plotted.</p>
<p>It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample.</p>
<p>An ideal input with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin etc.) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.</p>
<p><embed src="multiBAM_fingerprint_dunnart_downSampled.pdf" width="600" height="500" /> <embed src="H3K4me3_multiBAM_fingerprint_mouse.pdf" width="600" height="500" /> <embed src="H3K27ac_multiBAM_fingerprint_mouse.pdf" width="600" height="500" /></p>
</div>
</div>
<div id="cross-correlation-analysis" class="section level2">
<h2>Cross-correlation analysis</h2>
<p>Information from: <a href="https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit" class="uri">https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit</a></p>
<p>This set of programs operate on mapped Illumina single-end read datasets in tagAlign or BAM format. Because my data is paired-end I need to only use the forward read.</p>
<p>A high-quality ChIP-seq experiment will produce significant clustering of enriched DNA sequence tags/reads at locations bound by the protein of interest; the expectation is that we can observe a bimodal enrichment of reads (sequence tags) on both the forward and the reverse strands.</p>
<p>Cross-correlation analysis is done on a filtered (but not-deduped) and subsampled BAM. There is a special fastq trimming for cross-correlation analysis. Read1 fastq is trimmed to 50bp first using trimfastq.py (last modified 2017/11/08, <a href="https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py" class="uri">https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py</a>). And then it is separately mapped as SE. Reads are filtered but duplicates are not removed. Then 15 million reads are randomly sampled and used for cross-correlation analysis.</p>
<p>ENCODE standards:</p>
<table>
<colgroup>
<col width="4%" />
<col width="14%" />
<col width="80%" />
</colgroup>
<thead>
<tr class="header">
<th>col.</th>
<th>abbreviation</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Filename</td>
<td>tagAlign/BAM filename</td>
</tr>
<tr class="even">
<td>2</td>
<td>numReads</td>
<td>effective sequencing depth i.e. total number of mapped reads in input file</td>
</tr>
<tr class="odd">
<td>3</td>
<td>estFragLen</td>
<td>comma separated strand cross-correlation peak(s) in decreasing order of correlation.</td>
</tr>
<tr class="even">
<td>4</td>
<td>corr_estFragLen</td>
<td>comma separated strand cross-correlation value(s) in decreasing order (COL2 follows the same order)</td>
</tr>
<tr class="odd">
<td>5</td>
<td>phantomPeak</td>
<td>Read length/phantom peak strand shift</td>
</tr>
<tr class="even">
<td>6</td>
<td>corr_phantomPeak</td>
<td>Correlation value at phantom peak</td>
</tr>
<tr class="odd">
<td>7</td>
<td>argmin_corr</td>
<td>strand shift at which cross-correlation is lowest</td>
</tr>
<tr class="even">
<td>8</td>
<td>min_corr</td>
<td>minimum value of cross-correlation</td>
</tr>
<tr class="odd">
<td>9</td>
<td>NSC</td>
<td>Normalized strand cross-correlation coefficient (NSC) = COL4 / COL8</td>
</tr>
<tr class="even">
<td>10</td>
<td>RSC</td>
<td>Relative strand cross-correlation coefficient (RSC) = (COL4 - COL8) / (COL6 - COL8)</td>
</tr>
<tr class="odd">
<td>11</td>
<td>QualityTag</td>
<td>Quality tag based on thresholded RSC (codes= -2:veryLow, -1:Low, 0:Medium, 1:High, 2:veryHigh)</td>
</tr>
</tbody>
</table>
<p>NSC; NSC&gt;1.1 (higher values indicate more enrichment; 1 = no enrichment)</p>
<p>RSC; RSC&gt;0.8 (0 = no signal; &lt;1 low quality ChIP; &gt;1 high enrichment</p>
<p>Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High; 2:veryHigh)</p>
<p>Table. Peak calling quality metrics for mouse ChIP-seq peaks after peak calling with MACS2</p>
</div>
<div id="peak-qc" class="section level2">
<h2>Peak QC</h2>
<div id="number-of-peaks" class="section level3">
<h3>Number of peaks</h3>
<div id="dunnart" class="section level4">
<h4>Dunnart</h4>
<pre class="bash"><code>echo &quot;Number of peaks A3 H3K27ac dunnart&quot;
less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks A2 H3K4me3 dunnart&quot;
less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks B3 H3K27ac dunnart&quot;
less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks B2 H3K4me3 dunnart&quot;
less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l</code></pre>
</div>
<div id="mouse" class="section level4">
<h4>Mouse</h4>
<p>H3K4me3</p>
<pre class="bash"><code>echo &quot;Number of peaks rep 1 E10.5 H3K4me3&quot;
less output/peaks/ENCFF124UYX_vs_ENCFF157KEH_E10.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E10.5 H3K4me3&quot;
less output/peaks/ENCFF045IPK_vs_ENCFF825AVI_E10.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E11.5 H3K4me3&quot;
less output/peaks/ENCFF760QYZ_vs_ENCFF184CUE_E11.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E11.5 H3K4me3&quot;
less output/peaks/ENCFF717QDV_vs_ENCFF376FGM_E11.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E12.5 H3K4me3&quot;
less output/peaks/ENCFF941QJZ_vs_ENCFF058AUT_E12.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E12.5 H3K4me3&quot;
less output/peaks/ENCFF182ZPF_vs_ENCFF203JQV_E12.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E13.5 H3K4me3&quot;
less output/peaks/ENCFF485UDC_vs_ENCFF117QRC_E13.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E13.5 H3K4me3&quot;
less output/peaks/ENCFF124TAB_vs_ENCFF248PGK_E13.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E14.5 H3K4me3&quot;
less output/peaks/ENCFF665QBJ_vs_ENCFF002HZV_E14.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E14.5 H3K4me3&quot;
less output/peaks/ENCFF724DMU_vs_ENCFF784ORI_E14.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E15.5 H3K4me3&quot;
less output/peaks/ENCFF401BKM_vs_ENCFF182XFG_E15.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E15.5 H3K4me3&quot;
less output/peaks/ENCFF258KCR_vs_ENCFF727QTS_E15.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l</code></pre>
<p>H3K27ac</p>
<pre class="bash"><code>echo &quot;Number of peaks rep 1 E10.5 H3K27ac&quot;
less output/peaks/ENCFF213EBC_vs_ENCFF157KEH_E10.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E10.5 H3K27ac&quot;
less output/peaks/ENCFF548BRR_vs_ENCFF825AVI_E10.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E11.5 H3K27ac&quot;
less output/peaks/ENCFF512SFE_vs_ENCFF184CUE_E11.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E11.5 H3K27ac&quot;
less output/peaks/ENCFF515PKL_vs_ENCFF376FGM_E11.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E12.5 H3K27ac&quot;
less output/peaks/ENCFF011NFM_vs_ENCFF058AUT_E12.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E12.5 H3K27ac&quot;
less output/peaks/ENCFF394TZN_vs_ENCFF203JQV_E12.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E13.5 H3K27ac&quot;
less output/peaks/ENCFF194ORC_vs_ENCFF117QRC_E13.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 2 E13.5 H3K27ac&quot;
less output/peaks/ENCFF290ZNF_vs_ENCFF248PGK_E13.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E14.5 H3K27ac&quot;
less output/peaks/ENCFF902HAR_vs_ENCFF002HZV_E14.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l 
echo &quot;Number of peaks rep 2 E14.5 H3K27ac&quot;
less output/peaks/ENCFF327VAO_vs_ENCFF784ORI_E14.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of peaks rep 1 E15.5 H3K27ac&quot;
less output/peaks/ENCFF707WKL_vs_ENCFF182XFG_E15.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l 
echo &quot;Number of peaks rep 2 E15.5 H3K27ac&quot;
less output/peaks/ENCFF584JFB_vs_ENCFF727QTS_E15.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l</code></pre>
</div>
</div>
<div id="consensus-peaksets-for-biological-replicates" class="section level3">
<h3>Consensus peaksets for biological replicates</h3>
<div id="number-of-consensus-peaks" class="section level4">
<h4>Number of consensus peaks</h4>
<div id="dunnart-1" class="section level5">
<h5>Dunnart</h5>
<pre class="bash"><code>echo &quot;Number of consensus H3K4me3 dunnart peaks&quot;
less output/peaks/H3K4me3_overlap_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac dunnart peaks&quot;
less output/peaks/H3K27ac_overlap_default_dunnart_downSampled_peaks.narrowPeak | wc -l</code></pre>
</div>
<div id="mouse-1" class="section level5">
<h5>Mouse</h5>
<pre class="bash"><code>echo &quot;Number of consensus H3K4me3 mouse E10.5 peaks&quot;
less output/peaks/H3K4me3_E10.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K4me3 mouse E11.5 peaks&quot;
less output/peaks/H3K4me3_E11.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K4me3 mouse E12.5 peaks&quot;
less output/peaks/H3K4me3_E12.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K4me3 mouse E13.5 peaks&quot;
less output/peaks/H3K4me3_E13.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K4me3 mouse E14.5 peaks&quot;
less output/peaks/H3K4me3_E14.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K4me3 mouse E15.5 peaks&quot;
less output/peaks/H3K4me3_E15.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E10.5 peaks&quot;
less output/peaks/H3K27ac_E10.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E11.5 peaks&quot;
less output/peaks/H3K27ac_E11.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E12.5 peaks&quot;
less output/peaks/H3K27ac_E12.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E13.5 peaks&quot;
less output/peaks/H3K27ac_E13.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E14.5 peaks&quot;
less output/peaks/H3K27ac_E14.5_overlap.narrowPeak | wc -l
echo &quot;Number of consensus H3K27ac mouse E15.5 peaks&quot;
less output/peaks/H3K27ac_E15.5_overlap.narrowPeak | wc -l</code></pre>
</div>
</div>
</div>
</div>
</div>
<div id="consensus-peaks-and-grouping-into-putative-enhancers-and-promoters" class="section level1">
<h1>Consensus peaks and grouping into putative enhancers and promoters</h1>
<p>Find uniquely H3K4me3 sites (i.e. peaks that don’t overlap with H3K27ac):</p>
<pre class="bash"><code>conda activate chip
bedtools intersect -v -a output/peaks/H3K4me3_E10.5_overlap.narrowPeak -b output/peaks/H3K27ac_E10.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E10.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E11.5_overlap.narrowPeak -b output/peaks/H3K27ac_E11.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E11.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E12.5_overlap.narrowPeak -b output/peaks/H3K27ac_E12.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E12.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E13.5_overlap.narrowPeak -b output/peaks/H3K27ac_E13.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E13.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E14.5_overlap.narrowPeak -b output/peaks/H3K27ac_E14.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E14.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E15.5_overlap.narrowPeak -b output/peaks/H3K27ac_E15.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_E15.5_only.narrowPeak


bedtools intersect -v -a output/peaks/H3K4me3_overlap_default_dunnart_downSampled_peaks.narrowPeak -b output/peaks/H3K27ac_overlap_default_dunnart_downSampled_peaks.narrowPeak &gt; output/peaks/H3K4me3_only_dunnart_downSampled.narrowPeak</code></pre>
<p>Find unique H3K27ac sites (i.e. peaks that don’t overlap with H3K4me3):</p>
<pre class="bash"><code>bedtools intersect -v -a output/peaks/H3K27ac_E10.5_overlap.narrowPeak -b output/peaks/H3K4me3_E10.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E10.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E11.5_overlap.narrowPeak -b output/peaks/H3K4me3_E11.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E11.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E12.5_overlap.narrowPeak -b output/peaks/H3K4me3_E12.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E12.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E13.5_overlap.narrowPeak -b output/peaks/H3K4me3_E13.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E13.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E14.5_overlap.narrowPeak -b output/peaks/H3K4me3_E14.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E14.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E15.5_overlap.narrowPeak -b output/peaks/H3K4me3_E15.5_overlap.narrowPeak &gt; output/peaks/H3K27ac_E15.5_only.narrowPeak


bedtools intersect -v -a output/peaks/H3K27ac_overlap_default_dunnart_downSampled_peaks.narrowPeak -b output/peaks/H3K4me3_overlap_default_dunnart_downSampled_peaks.narrowPeak &gt; output/peaks/H3K27ac_only_dunnart_downSampled.narrowPeak</code></pre>
<p>Find peaks common between H3K27ac &amp; H3K4me3 with a reciprocal overlap of at least 50%.</p>
<pre class="bash"><code>bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E10.5_overlap.narrowPeak -b output/peaks/H3K27ac_E10.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E10.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E11.5_overlap.narrowPeak -b output/peaks/H3K27ac_E11.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E11.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E12.5_overlap.narrowPeak -b output/peaks/H3K27ac_E12.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E12.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E13.5_overlap.narrowPeak -b output/peaks/H3K27ac_E13.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E13.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E14.5_overlap.narrowPeak -b output/peaks/H3K27ac_E14.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E14.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E15.5_overlap.narrowPeak -b output/peaks/H3K27ac_E15.5_overlap.narrowPeak &gt; output/peaks/H3K4me3_and_H3K27ac_E15.5.narrowPeak

bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_overlap_default_dunnart_downSampled_peaks.narrowPeak -b output/peaks/H3K27ac_overlap_default_dunnart_downSampled_peaks.narrowPeak &gt; output/peaks/H3K4me3_H3K27ac_dunnart_downSampled.narrowPeak</code></pre>
<p>Combine peak files</p>
<pre class="bash"><code>cat output/peaks/H3K4me3_E10.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E10.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E10.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E11.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E11.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E11.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E12.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E12.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E12.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E13.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E13.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E13.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E14.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E14.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E14.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E15.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E15.5.narrowPeak | sort -k 4,4 &gt; output/peaks/E15.5_promoter_peaks.narrowPeak

cat output/peaks/H3K4me3_only_dunnart_downSampled.narrowPeak output/peaks/H3K4me3_H3K27ac_dunnart_downSampled.narrowPeak| sort -k 4,4 &gt; output/peaks/dunnart_downSampled_promoter_peaks.narrowPeak


cat output/peaks/H3K27ac_E10.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E10.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E11.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E11.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E12.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E12.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E13.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E13.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E14.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E14.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E15.5_only.narrowPeak | sort -k 4,4 &gt; output/peaks/E15.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_only_dunnart_downSampled.narrowPeak | sort -k 4,4 &gt; output/peaks/dunnart_downSampled_enhancer_peaks.narrowPeak</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS/LAPACK: /usr/local/easybuild-2019/easybuild/software/compiler/gcc/10.2.0/openblas/0.3.12/lib/libopenblas_haswellp-r0.3.12.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.33        forcats_0.5.1     readr_1.4.0       tidyr_1.1.3      
 [5] tibble_3.1.2      ggplot2_3.3.3     tidyverse_1.3.1   purrr_0.3.4      
 [9] stringr_1.4.0     dplyr_1.0.8       data.table_1.14.0 workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        lubridate_1.7.10  getPass_0.2-2     ps_1.6.0         
 [5] assertthat_0.2.1  rprojroot_2.0.2   digest_0.6.27     utf8_1.2.1       
 [9] R6_2.5.0          cellranger_1.1.0  backports_1.2.1   reprex_2.0.0     
[13] evaluate_0.14     highr_0.9         httr_1.4.2        pillar_1.6.1     
[17] rlang_1.0.2       readxl_1.3.1      rstudioapi_0.13   whisker_0.4      
[21] callr_3.7.0       jquerylib_0.1.4   rmarkdown_2.8     munsell_0.5.0    
[25] broom_0.7.6       compiler_4.1.0    httpuv_1.6.1      modelr_0.1.8     
[29] xfun_0.23         pkgconfig_2.0.3   htmltools_0.5.1.1 tidyselect_1.1.1 
[33] fansi_0.5.0       crayon_1.4.1      dbplyr_2.1.1      withr_2.4.2      
[37] later_1.2.0       grid_4.1.0        jsonlite_1.7.2    gtable_0.3.0     
[41] lifecycle_1.0.1   DBI_1.1.1         git2r_0.28.0      magrittr_2.0.1   
[45] scales_1.1.1      cli_2.5.0         stringi_1.6.2     fs_1.5.0         
[49] promises_1.2.0.1  xml2_1.3.2        bslib_0.2.5.1     ellipsis_0.3.2   
[53] generics_0.1.0    vctrs_0.3.8       tools_4.1.0       glue_1.4.2       
[57] hms_1.1.0         processx_3.5.2    yaml_2.2.1        colorspace_2.0-1 
[61] rvest_1.0.0       haven_2.4.1       sass_0.4.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
