---
title: "Dunnart ChIP-seq data pre-processing"
author: "lecook"
date: "2022-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# Snakemake pipeline
For full snakemake script see code/dunnart_peak_calling/dunnart_snakefile.

## Create conda environment

```{bash eval=FALSE}
# create conda environment with all packages and dependencies
conda create --name chip bowtie2 samtools picard deeptools multiqc phantompeakqualtools preseq macs2 bedtools fastqc pybedtools

# save environment to a yaml file
conda env export > chip_environment.yml

# activate environment before running pipeline
conda activate chip
```

## Run snakemake from command line
Cluster configuration file: code/configs/cluster.json
Sample configuration file: code/configs/config.yaml
Sample text file: code/configs/SSR.text

```{bash eval=FALSE}
module load snakemake/6.6.1
snakemake --snakemakefile code/dunnart_peak_calling/dunnart_snakefile -j 6 --cluster-config envs/cluster.json --cluster "sbatch -A {cluster.account} -t {cluster.time} -p {cluster.partition} --nodes {cluster.nodes} --ntasks {cluster.ntasks} --mem {cluster.mem}" &
```

## Effective genome length
We can approximate effective genome size for various read lengths using the khmer program and `unique-kmers.py`. This will estimate the number of unique kmers (for a specified length kmer) which can be used to infer the total uniquely mappable genome. (I.e it doesn't include highly repetitive regions). https://khmer.readthedocs.io/en/v2.1.1/user/scripts.html
This was a suggestion of deepTools: https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html

```{bash eval=FALSE}
module load pip/21.2.4-python-3.8.6
pip2.7 install khmer
```

Run `unique-kmers.py` on dunnart genome for read length of 150bp:


```{bash eval=FALSE}
/usr/local/bin/unique-kmers.py -k 150 Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa
```
```
Estimated number of unique 150-mers in /Users/lauracook/../../Volumes/macOS/genomes/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fasta: 2740338543
Total estimated number of unique 150-mers: 2740338543
```

## Indexing genome file

Load modules:

```{bash eval=FALSE}
module load gcc/8.3.0
module load bowtie2/2.3.5.1
```

Build index:
```{bash eval=FALSE}
bowtie2-build Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr
```

## Align reads and filter using samtools and Picard 
See full snakemake script for details: code/dunnart_peak_calling/dunnart_snakefile.

|  <br>                                                                                |               | **read counts**                                             |                    |                      |            |             |              |
| :----------------------------------------------------------------------------------: | :-----------: | :---------------------------------------------------------: | :----------------: | :------------------: | :--------: | :---------: | :----------: |
| **Sample**                                                                           | **Antibody**  | *bowtie2 alignment*                                         | *filtered reads\^*  | *remove duplicates*  | **NRF\***  | **PBC1\***  | **PBC2\***   |
| A-1                                                                                  | input         | 130587134                                                   | 87165286           | 47645260             | 0\.773602  | 0\.774519   | 4\.45728     |
| A-2                                                                                  | H3K4me3       | 103324742                                                   | 62678360           | 35617708             | 0\.789709  | 0\.789661   | 4\.754989    |
| A-3                                                                                  | H3K27ac       | 131071676                                                   | 86503078           | 45700530             | 0\.761768  | 0\.761687   | 4\.196586    |
| B-1                                                                                  | input         | 111574640                                                   | 72303316           | 44433864             | 0\.820313  | 0\.819974   | 5\.545324    |
| B-2                                                                                  | H3K4me3       | 114146802                                                   | 67048994           | 42357180             | 0\.831032  | 0\.830749   | 5\.898224    |
| B-3                                                                                  | H3K27ac       | 104714846                                                   | 66615704           | 43939460             | 0\.847255  | 0\.847022   | 6\.529168    |
|                                                                                      |               |                                                             |                    |                      |            |             |              |
| \* calculated from a subsample of ~10-15M aligned reads prior to removing duplicates |               |                                                             |                    |                      |            |             |              |
| Ì‚ excluded low quality (MAPQ  30) and orphaned reads                                 |               |                                                             |                    |                      |            |             |              |
