---
title: "Dunnart ChIP-seq data pre-processing"
author: "lecook"
date: "2022-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# Snakemake pipeline
For full snakemake script see code/dunnart_peak_calling/dunnart_snakefile.

## Create conda environment

```{bash eval=FALSE}
# create conda environment with all packages and dependencies
conda create --name chip bowtie2 samtools picard deeptools multiqc phantompeakqualtools preseq macs2 bedtools fastqc snakemake numpy pybedtools

# save environment to a yaml file
conda env export > chip_environment.yml

# activate environment before running pipeline
conda activate chip
```

## Run snakemake from command line

Cluster configuration file: code/configs/cluster.json

Sample configuration file: code/configs/config.yaml

Sample text file: code/configs/SSR.text

```{bash eval=FALSE}
module load snakemake/6.6.1
snakemake --snakemakefile code/dunnart_peak_calling/dunnart_snakefile -j 6 --cluster-config envs/cluster.json --cluster "sbatch -A {cluster.account} -t {cluster.time} -p {cluster.partition} --nodes {cluster.nodes} --ntasks {cluster.ntasks} --mem {cluster.mem}" &
```

## Effective genome length
We can approximate effective genome size for various read lengths using the khmer program and `unique-kmers.py`. This will estimate the number of unique kmers (for a specified length kmer) which can be used to infer the total uniquely mappable genome. (I.e it doesn't include highly repetitive regions). https://khmer.readthedocs.io/en/v2.1.1/user/scripts.html

This was a suggestion of deepTools: https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html

```{bash eval=FALSE}
module load pip/21.2.4-python-3.8.6
pip2.7 install khmer
```

Run `unique-kmers.py` on dunnart genome for read length of 150bp:


```{bash eval=FALSE}
/usr/local/bin/unique-kmers.py -k 150 Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa
```
```
Estimated number of unique 150-mers in /Users/lauracook/../../Volumes/macOS/genomes/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fasta: 2740338543
Total estimated number of unique 150-mers: 2740338543
```

## Indexing genome file

Load modules:

```{bash eval=FALSE}
module load gcc/8.3.0
module load bowtie2/2.3.5.1
```

Build index:
```{bash eval=FALSE}
bowtie2-build Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr.fa Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr
```

## Align and filter dunnart reads

Uses samtools and picards MarkDuplicates to remove low quality reads. This is based on and in accordance with the ENCODE Guidelines and pipeline (https://github.com/ENCODE-DCC/chip-seq-pipeline2)

__Library Complexity ChIP-seq Standards:__

| PBC1 | PBC2 | Bottlenecking level | NRF | Complexity | Flag colors |
|:-:|:-:|:-:|:-:|:-:|:-:|
| < 0.5 | < 1 | Severe | < 0.5 | Concerning | Orange |
| 0.5 ≤ PBC1 < 0.8 | 1 ≤ PBC2 < 3 | Moderate | 0.5 ≤ NRF < 0.8 | Acceptable | Yellow |
| 0.8 ≤ PBC1 < 0.9 | 3 ≤ PBC2 < 10 | Mild | 0.8 ≤ NRF < 0.9 | Compliant | None |
| ≥ 0.9 | ≥ 10 | None | > 0.9 | Ideal | None |

See full snakemake script for details: code/dunnart_peak_calling/dunnart_snakefile.

Table. Read alignment quality metrics before and aftering filtering steps
|  <br>            |               | **read counts**        |                    |                      |            |             |              |
| :--------------: | :-----------: | :--------------------: | :----------------: | :------------------: | :--------: | :---------: | :----------: |
| **Sample**       | **Antibody**  | *bowtie2 alignment*    | *filtered reads\^* | *remove duplicates*  | **NRF\***  | **PBC1\***  | **PBC2\***   |
| A-1              | input         | 130587134              | 87165286           | 47645260             | 0\.773602  | 0\.774519   | 4\.45728     |
| A-2              | H3K4me3       | 103324742              | 62678360           | 35617708             | 0\.789709  | 0\.789661   | 4\.754989    |
| A-3              | H3K27ac       | 131071676              | 86503078           | 45700530             | 0\.761768  | 0\.761687   | 4\.196586    |
| B-1              | input         | 111574640              | 72303316           | 44433864             | 0\.820313  | 0\.819974   | 5\.545324    |
| B-2              | H3K4me3       | 114146802              | 67048994           | 42357180             | 0\.831032  | 0\.830749   | 5\.898224    |
| B-3              | H3K27ac       | 104714846              | 66615704           | 43939460             | 0\.847255  | 0\.847022   | 6\.529168    |

\* calculated from a subsample of ~10-15M aligned reads prior to removing duplicates

^ excluded low quality (MAPQ  30) and orphaned reads

## Read QC with deepTools

### Plot correlation between BAM files
`plotCorrelation` computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions.

This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.

![Pearson correlation dunnart BAM files](output/qc/pearsoncor_multibamsum.png)


### Fingerprint plots 
This tool samples indexed BAM files and plots a profile of cumulative read coverages for each. All reads overlapping a window (bin) of the specified length are counted; these counts are sorted and the cumulative sum is finally plotted.

It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample.

An ideal input with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin etc.) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.

![Fingerprint plot for dunnart BAM files](output/qc/multiBAM_fingerprint.png)

### BAM PE fragment size

This tool calculates the fragment sizes for read pairs given a BAM file from paired-end sequencing. Several regions are sampled depending on the size of the genome and number of processors to estimate thesummary statistics on the fragment lengths.

![Fragment size for read pairs in a BAM file](output/qc/bamPEFragmentSize_hist.png)

## Cross-correlation analysis
Information from: https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit

This set of programs operate on mapped Illumina single-end read datasets in tagAlign or BAM format. Because my data is paired-end I need to only use the forward read.

A high-quality ChIP-seq experiment will produce significant clustering of enriched DNA sequence tags/reads at locations bound by the protein of interest; the expectation is that we can observe a bimodal enrichment of reads (sequence tags) on both the forward and the reverse strands.

Cross-correlation analysis is done on a filtered (but not-deduped) and subsampled BAM. There is a special fastq trimming for cross-correlation analysis. Read1 fastq is trimmed to 50bp first using trimfastq.py (last modified 2017/11/08, https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py). And then it is separately mapped as SE. Reads are filtered but duplicates are not removed. Then 15 million reads are randomly sampled and used for cross-correlation analysis.

ENCODE standards:

| col. | abbreviation     | description                                                                                         |
|------|------------------|-----------------------------------------------------------------------------------------------------|
| 1    | Filename         | tagAlign/BAM filename                                                                               |
| 2    | numReads         | effective sequencing depth i.e. total number of mapped reads in input file                          |
| 3    | estFragLen       | comma separated strand cross-correlation peak(s) in decreasing order of correlation.                |
| 4    | corr_estFragLen  | comma separated strand cross-correlation value(s) in decreasing order (COL2 follows the same order) |
| 5    | phantomPeak      | Read length/phantom peak strand shift                                                               |
| 6    | corr_phantomPeak | Correlation value at phantom peak                                                                   |
| 7    | argmin_corr      | strand shift at which cross-correlation is lowest                                                   |
| 8    | min_corr         | minimum value of cross-correlation                                                                  |
| 9    | NSC              | Normalized strand cross-correlation coefficient (NSC) = COL4 / COL8                                 |
| 10   | RSC              | Relative strand cross-correlation coefficient (RSC) = (COL4 - COL8) / (COL6 - COL8)                 |
| 11   | QualityTag       | Quality tag based on thresholded RSC (codes= -2:veryLow, -1:Low, 0:Medium, 1:High, 2:veryHigh)      |


NSC; NSC>1.1 (higher values indicate more enrichment; 1 = no enrichment)

RSC; RSC>0.8 (0 = no signal; <1 low quality ChIP; >1 high enrichment

Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High; 2:veryHigh)

```{r}
library(data.table)
library(knitr)
A1 = fread("output/qc/A-1_input_filt_15Mreads.SE.cc.qc")
colnames(A1) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
A2 = fread("output/qc/A-2_H3K4me3_filt_15Mreads.SE.cc.qc")
colnames(A2) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
A3 = fread("output/qc/A-3_H3K27ac_filt_15Mreads.SE.cc.qc")
colnames(A3) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
B1 = fread("output/qc/B-1_input_filt_15Mreads.SE.cc.qc")
colnames(B1) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
B2 = fread("output/qc/B-2_H3K4me3_filt_15Mreads.SE.cc.qc")
colnames(B2) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
B3 = fread("output/qc/B-3_H3K27ac_filt_15Mreads.SE.cc.qc")
colnames(B3) = c("Filename", "numReads", "estFragLen", "corr_estFragLen", "PhantomPeak", "corr_phantomPeak", "argmin_corr", "min_corr", "phantomPeakCoef", "relPhantomPeakCoef", "QualityTag")
```


```{r echo=FALSE}
kable(A1, caption="A1 H3K27ac cross-correlation analysis")
```


```{r echo=FALSE}
kable(A2, caption="A2 H3K4me3 cross-correlation analysis")
```

```{r echo=FALSE}
kable(A3, caption="A3 input control cross-correlation analysis")
```

```{r echo=FALSE}
kable(B1, caption="B1 H3K27ac control cross-correlation analysis")
```

```{r echo=FALSE}
kable(B2, caption="B2 H3K4me3 control cross-correlation analysis")
```

```{r echo=FALSE}
kable(B3, caption="B3 input control cross-correlation analysis")
```

## Peak QC 

Number of peaks 

```{bash}
less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l
```

```{bash}
less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l
```

```{bash}
less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l
```

```{bash}
less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l
```

### Fraction of reads in peaks

```{r}
A2 = fread("output/qc/A-2_H3K4me3_vs_A-1_input.frip_default.txt")
A3 = fread("output/qc/A-3_H3K27ac_vs_A-1_input.frip_default.txt")
B2 = fread("output/qc/B-2_H3K4me3_vs_B-1_input.frip_default.txt")
B3 = fread("output/qc/B-3_H3K27ac_vs_B-1_input.frip_default.txt")
```

A-2_H3K4me3 fraction of reads in peaks
```{bash}
less output/qc/A-2_H3K4me3_vs_A-1_input.frip_default.txt)
```

A3 H3K27ac fraction of reads in peaks
```{bash}
less output/qc/A-3_H3K27ac_vs_A-1_input.frip_default.txt)
```

B2 H3K4me3 fraction of reads in peaks
```{bash}
less output/qc/B-2_H3K4me3_vs_B-1_input.frip_default.txt
```

B3 H3K27ac fraction of reads in peaks
```{bash}
less output/qc/B-3_H3K27ac_vs_B-1_input.frip_default.txt
```

### Consensus peaksets for biological replicates
Number of consensus peaks 

```{bash}
less output/peaks/H3K4me3_overlap_default.narrowPeak | wc -l
```

```{bash}
less output/peaks/H3K27ac_overlap_default.narrowPeak | wc -l
```

## Group peaks into promoter- and enhancer- associated peaks

Find uniquely H3K4me3 sites (i.e. peaks that don't overlap with H3K27ac):

```{bash}
bedtools intersect -v -a output/peaks/H3K4me3_overlap.narrowPeak -b output/peaks/H3K27ac_overlap.narrowPeak > output/peaks/H3K4me3_only.narrowPeak
less output/peaks/H3K4me3_only.narrowPeak | wc -l
```

Find unique H3K27ac sites (i.e. peaks that don't overlap with H3K4me3):
```{bash}
bedtools intersect -v -a output/peaks/H3K27ac_overlap.narrowPeak -b output/peaks/H3K4me3_overlap.narrowPeak > output/peaks/H3K27ac_only.narrowPeak
less output/peaks/H3K27ac_only.narrowPeak | wc -l
```

Find peaks common between H3K27ac & H3K4me3 with a reciprocal overlap of at least 50%.
```{bash}
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_overlap.narrowPeak -b output/peaks/H3K27ac_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac.narrowPeak
less output/peaks/H3K4me3_and_H3K27ac.narrowPeak | wc -l
```

Combine files

```{bash}
cat output/peaks/H3K4me3_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac.narrowPeak | sort -k 4,4 > output/peaks/dunnart_promoter_peaks.narrowPeak
cat output/peaks/H3K27ac_only.narrowPeak | sort -k 4,4 > output/peaks/dunnart_enhancer_peaks.narrowPeak
less output/peaks/dunnart_promoter_peaks.narrowPeak | wc -l
less output/peaks/dunnart_enhancer_peaks.narrowPeak | wc -l
```
