---
title: "Mouse and dunnart peak features"
author: "lecook"
date: "2022-03-16"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Set-up
```{r setup, include = FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# Load in libraries
library(ChIPseeker)
library(GenomicFeatures)
library(ggplot2)
library(data.table)
library(dplyr)
library(TxDb.Mmusculus.UCSC.mm10.ensGene)
library(scales)

plot_dir <- "output/plots/"
fullPeak_dir <- "output/peaks/"
annot_dir <- "output/annotations/"
filterPeaks_dir <- "output/filtered_peaks/"

## Set the fonts up so that each plot is the saved the same way.
font <- theme(axis.text.x = element_text(size = 25),
        axis.text.y = element_text(size = 25),
        axis.title.x = element_text(size = 25),
        axis.title.y = element_text(size = 25), 
        legend.title = element_text(size = 25), legend.text = element_text(size = 25))
```


# Peak features
## Peak counts reads normalised to 10 million reads
```{r}
files =list.files(fullPeak_dir, pattern= "*downSampled.narrowPeak", full.names=T) # create list of files in directory
files = as.list(files)
data = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA")))


data[[1]]$mark = "H3K27ac"
data[[2]]$mark = "H3K27ac"
data[[3]]$mark = "H3K27ac"
data[[4]]$mark = "H3K27ac"
data[[5]]$mark = "H3K27ac"
data[[6]]$mark = "H3K27ac"
data[[7]]$mark = "H3K27ac"
data[[8]]$mark = "H3K27ac_H3K4me3"
data[[9]]$mark = "H3K27ac_H3K4me3"
data[[10]]$mark = "H3K27ac_H3K4me3"
data[[11]]$mark = "H3K27ac_H3K4me3"
data[[12]]$mark = "H3K27ac_H3K4me3"
data[[13]]$mark = "H3K27ac_H3K4me3"
data[[14]]$mark = "H3K4me3"
data[[15]]$mark = "H3K4me3"
data[[16]]$mark = "H3K4me3"
data[[17]]$mark = "H3K4me3"
data[[18]]$mark = "H3K4me3"
data[[19]]$mark = "H3K4me3"
data[[20]]$mark = "H3K27ac_H3K4me3"
data[[21]]$mark = "H3K4me3"


data[[1]]$sample = "E10.5"
data[[2]]$sample = "E11.5"
data[[3]]$sample = "E12.5"
data[[4]]$sample = "E13.5"
data[[5]]$sample = "E14.5"
data[[6]]$sample = "E15.5"
data[[7]]$sample = "dunnart"
data[[8]]$sample = "E10.5"
data[[9]]$sample = "E11.5"
data[[10]]$sample = "E12.5"
data[[11]]$sample = "E13.5"
data[[12]]$sample = "E14.5"
data[[13]]$sample = "E15.5"
data[[14]]$sample = "E10.5"
data[[15]]$sample = "E11.5"
data[[16]]$sample = "E12.5"
data[[17]]$sample = "E13.5"
data[[18]]$sample = "E14.5"
data[[19]]$sample = "E15.5"
data[[20]]$sample = "dunnart"
data[[21]]$sample = "dunnart"

## Plot stacked bar graph with number of peaks for each
combined.peaks <- rbindlist(data,)
combined.tbl <- with(combined.peaks, table(sample, mark))
combined.tbl <- as.data.frame(combined.tbl)

p <- ggplot(combined.tbl, aes(factor(sample), Freq, fill=mark)) + 
  geom_bar(position=position_stack(reverse = TRUE), stat="identity") +
  theme_minimal() + ylab("Number of peaks") +
  xlab("") + scale_color_manual(values = c("#ffd166", "#299d8f", "#244653")) + scale_fill_manual(values = c("#ffd166", "#299d8f", "#244653")) +
  theme_bw()

p

pdf(file=paste0(plot_dir, "number_of_mouse_dunnart_peaks.pdf"), width = 10, height = 7)
print(p + font)
dev.off()
```

## Peak counts ALL reads
```{r}
files =list.files(fullPeak_dir, pattern= "*5.narrowPeak|*only.narrowPeak|*ac.narrowPeak", full.names=T) # create list of files in directory
files = as.list(files)
data = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA")))


data[[1]]$mark = "H3K27ac"
data[[2]]$mark = "H3K27ac"
data[[3]]$mark = "H3K27ac"
data[[4]]$mark = "H3K27ac"
data[[5]]$mark = "H3K27ac"
data[[6]]$mark = "H3K27ac"
data[[7]]$mark = "H3K27ac"
data[[8]]$mark = "H3K27ac_H3K4me3"
data[[9]]$mark = "H3K27ac_H3K4me3"
data[[10]]$mark = "H3K27ac_H3K4me3"
data[[11]]$mark = "H3K27ac_H3K4me3"
data[[12]]$mark = "H3K27ac_H3K4me3"
data[[13]]$mark = "H3K27ac_H3K4me3"
data[[14]]$mark = "H3K27ac_H3K4me3"
data[[15]]$mark = "H3K4me3"
data[[16]]$mark = "H3K4me3"
data[[17]]$mark = "H3K4me3"
data[[18]]$mark = "H3K4me3"
data[[19]]$mark = "H3K4me3"
data[[20]]$mark = "H3K4me3"
data[[21]]$mark = "H3K4me3"


data[[1]]$sample = "E10.5"
data[[2]]$sample = "E11.5"
data[[3]]$sample = "E12.5"
data[[4]]$sample = "E13.5"
data[[5]]$sample = "E14.5"
data[[6]]$sample = "E15.5"
data[[7]]$sample = "dunnart"
data[[8]]$sample = "E10.5"
data[[9]]$sample = "E11.5"
data[[10]]$sample = "E12.5"
data[[11]]$sample = "E13.5"
data[[12]]$sample = "E14.5"
data[[13]]$sample = "E15.5"
data[[14]]$sample = "dunnart"
data[[15]]$sample = "E10.5"
data[[16]]$sample = "E11.5"
data[[17]]$sample = "E12.5"
data[[18]]$sample = "E13.5"
data[[19]]$sample = "E14.5"
data[[20]]$sample = "E15.5"
data[[21]]$sample = "dunnart"

## Plot stacked bar graph with number of peaks for each
combined.peaks <- rbindlist(data,)
combined.tbl <- with(combined.peaks, table(sample, mark))
combined.tbl <- as.data.frame(combined.tbl)

p <- ggplot(combined.tbl, aes(factor(sample), Freq, fill=mark)) + 
  geom_bar(position=position_stack(reverse = TRUE), stat="identity") +
  theme_minimal() + ylab("Number of peaks") + scale_y_continuous(labels = comma) +
  xlab("") + scale_color_manual(values = c("#ffd166", "#299d8f", "#244653")) + scale_fill_manual(values = c("#ffd166", "#299d8f", "#244653")) +
  theme_bw() 


p + ggtitle("Number of peaks with all reads used for peak calling")

pdf(file=paste0(plot_dir, "number_of_mouse_dunnart_peaks_all_reads.pdf"), width = 10, height = 7)
print(p + font)
dev.off()
```

Use peaks called with all reads for downstream analyses as the enrichment in the mouse data is weaker. 

## Peak lengths for H3K4me3 and H3K27ac
```{r}
files =list.files(fullPeak_dir, pattern= "*5_enhancer_peaks.narrowPeak|*5_promoter_peaks.narrowPeak|dunnart_enhancer_peaks.narrowPeak|dunnart_promoter_peaks.narrowPeak", full.names=T) # create list of files in directory
filenames <- sub('\\_peaks.narrowPeak$', '', basename(files)) 
files = as.list(files)
data = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA")))
names(data) <- filenames

df1 = Map(mutate, data[c(1,3,5,7,9,11,13)], cre = "enhancer")
df2 = Map(mutate, data[c(2,4,6,8,10,12,14)], cre = "promoter")
data = append(df1, df2)

df1 = Map(mutate, data[c(1,8)], group = "dunnart")
df2 = Map(mutate, data[c(2,9)], group = "E10.5")
df3 = Map(mutate, data[c(3,10)], group = "E11.5")
df4 = Map(mutate, data[c(4,11)], group = "E12.5")
df5 = Map(mutate, data[c(5,12)], group = "E13.5")
df6 = Map(mutate, data[c(6,13)], group = "E14.5")
df7 = Map(mutate, data[c(7,14)], group = "E15.5")

data <- append(df1, df2)
data <- append(data, df3)
data = append(data, df4)
data = append(data, df5)
data = append(data, df6)
data = append(data, df7)

data = rbindlist(data,)
data$length = data$V3 - data$V2

p = ggplot(data, aes(x=factor(group), y=log10(length), fill = cre)) + geom_violin(aes(fill=factor(cre)),
              position = position_dodge(width=0.8)) + 
  geom_boxplot(aes(fill=factor(cre)), 
               width=.2,
               outlier.shape = NA,
               notch=FALSE,
               position = position_dodge(width=0.8)) +
  theme_bw() + xlab("") + ylab("Log10 Peak Length") +  scale_color_manual(values = c("#efc769", "#1a6259")) +
    scale_fill_manual(values = c("#ffe29f","#8db1ac"))
p
pdf(file=paste0(plot_dir, "mouse_dunnart_peak_lengths.pdf"), width = 10, height = 7)
print(p  + font)
dev.off()
```


## Plot peak intensity
```{r}
p = ggplot(data, aes(x=factor(group), y=log10(V7), fill = cre)) + geom_violin(aes(fill=factor(cre)),
              position = position_dodge(width=0.8)) + 
  geom_boxplot(aes(fill=factor(cre)), 
               width=.2,
               outlier.shape = NA,
               notch=FALSE,
               position = position_dodge(width=0.8)) +
  theme_bw() + xlab("") + ylab("Log10 Peak Intensity") +  scale_color_manual(values = c("#efc769", "#1a6259")) +
    scale_fill_manual(values = c("#ffe29f","#8db1ac"))
p
pdf(file=paste0(plot_dir, "mouse_dunnart_peak_intensity.pdf"), width = 10, height = 7)
print(p  + font)
dev.off()
```

# Annotate dunnart peaks
The easiest way to call the nearest genes for the peaks in the dunnart is to use the ChIPseeker package (Guangchuang Yu 2021) as it allows easy integration of non-model organism genomes and has well documented instructions on incorporating BYO genomes with the package.
To use the ChIPseeker to annotate peaks, firstly a txdb is needed for the dunnart annotation file. A TxDb class is a container for storing transcript annotations. The dunnart genome doesn't have a de novo annotation so instead the Tasmanian devil annotation (RefSeq) has been lifted over to the dunnart genome using LiftOff (https://github.com/agshumate/Liftoff). 

## Gene ID conversion tables
For downstream analyses, conversion tables between gene databases and between species is needed.
This is because the ENSEMBL/ENTREZ IDs for the Tasmanian Devil have fewer links to databases such as GO terms etc.
For this I have two conversion tables: 
1. Converts Tasmanian devil RefSeq to Tasmanian Devil ENSEMBL IDs
2. Convert Tasmanian Devil ENSEMBL IDs to mouse ENSEMBL IDs

Additionally I have a list of background genes for calculating GO enrichment. This background list includes all devil genes that have an orthologous mouse gene ID in the ensembl database.

## Annotation files
```{r}
# Using ENSEMBL version 103 for both the mouse and devil to keep it consistent
## Annotation file for the mouse

#mm10_txdb <- makeTxDbFromBiomart(biomart="ENSEMBL_MART_ENSEMBL",
#                    dataset="mmusculus_gene_ensembl",
#                    host="http://feb2021.archive.ensembl.org")
#seqlevelsStyle(mm10_txdb) <- "UCSC"
mm10_txdb <- TxDb.Mmusculus.UCSC.mm10.ensGene

## Make txdb for dunnart annotation file
smiCra_txdb <- makeTxDbFromGFF("data/genomic_data/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr2.gff")

## Convert geneIDs
### Tables downloaded from biomart and collated
df2 <- read.table("output/annotations/devil_to_mouse_ensembl.txt", header=TRUE, sep="\t") ## conversion table for devil ENSEMBL to mouse ENSEMBL
df3 <- read.table("output/annotations/refseq_to_ensembl.txt", header=TRUE, sep="\t") ## convertsion table for devil refseq to devil ENSEMBL
bg = fread("output/annotations/go_background_orthologousENSEMBL_biomart.txt", header = FALSE)
bg = unlist(bg, use.names = FALSE)
```

## Annotate peak files with ChIPseeker
### Mouse

```{r}
## Anotate peak files
annotatePeaksmm10 <- function(peak, outFile){
  
  # Annotate peak file based on mouse ENSEMBL annotation
  peakAnno <- annotatePeak(peak, tssRegion = c(-3000, 3000), TxDb = mm10_txdb)
  
  # Write annotation to file
  write.table(peakAnno, outFile, sep="\t", quote=F, row.names=F)
}

annotatePeaksmm10(peak = paste0(fullPeak_dir, "E10.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir, "E10.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E11.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir, "E11.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E12.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir, "E12.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E13.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir, "E13.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E14.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir, "E14.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E15.5_enhancer_peaks.narrowPeak"), outFile = paste0(annot_dir,"E15.5_enhancer_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E10.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E10.5_promoter_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E11.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E11.5_promoter_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E12.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E12.5_promoter_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E13.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E13.5_promoter_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E14.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E14.5_promoter_annotation.txt"))
annotatePeaksmm10(peak = paste0(fullPeak_dir, "E15.5_promoter_peaks.narrowPeak"), outFile = paste0(annot_dir,"E15.5_promoter_annotation.txt"))
```

### Dunnart
```{r}
annotatePeaks <- function(peak, outFile, outFile1, outFile2, GOenrich, kegg, backg){
  
  # Annotate peak file based on dunnart GFF
  peakAnno <- ChIPseeker::annotatePeak(peak = peak, tssRegion = c(-3000, 3000), TxDb = smiCra_txdb)

  # Write annotation to file
  write.table(peakAnno, file = paste0(annot_dir, outFile), sep = "\t", quote = F, row.names = F)
  
  peakAnnoDF <- as.data.frame(peakAnno, row.names = NULL)
  # # Convert refseq IDs and geneIDs to devil ensembl IDs
  df2 <- read.table("output/annotations/devil_to_mouse_ensembl.txt", header=TRUE, sep="\t") ## conversion table for devil ENSEMBL to mouse ENSEMBL
  df3 <- read.table("output/annotations/refseq_to_ensembl.txt", header=TRUE, sep="\t") ## convertsion table for devil refseq to devil ENSEMBL

  peakAnnoDF$ensemblgeneID <- df2$Gene.stable.ID[match(unlist(peakAnnoDF$geneId), df2$Gene.name)]
  peakAnnoDF$ensemblgeneID <- replace(peakAnnoDF$ensemblgeneID,is.na(peakAnnoDF$ensemblgeneID),"-")
  peakAnnoDF$transcriptIdAltered <- gsub("\\..*","", peakAnnoDF$transcriptId)
  peakAnnoDF$refseqID <- df3$Ensembl.Gene.ID[match(unlist(peakAnnoDF$transcriptIdAltered), df3$RefSeq.mRNA.Accession)]
  peakAnnoDF$refseqID <- replace(peakAnnoDF$refseqID,is.na(peakAnnoDF$refseqID),"-")
  peakAnnoDF$combined <- ifelse(peakAnnoDF$refseqID == "-", peakAnnoDF$ensemblgeneID, peakAnnoDF$refseqID)
  peakAnnoDF$combined[peakAnnoDF$combined == as.character("-")] <- NA

  peakAnnoDF <- peakAnnoDF[!is.na(peakAnnoDF$combined),]
  write.table(peakAnnoDF, paste0(annot_dir, outFile2), sep = "\t", quote = F, row.names = F)
  # # Convert devil ensembl to mouse ensembl
  peakAnnoDF$mouseensembl <- df2$Mouse.gene.stable.ID[match(unlist(peakAnnoDF$combined), df2$Gene.stable.ID)]
  # # Write annotation with converted IDs 
  peakAnnoDF$mouseensembl[peakAnnoDF$mouseensembl == as.character("")] <- NA
  peakAnnoDF <- peakAnnoDF[!is.na(peakAnnoDF$mouseensembl),]
  write.table(peakAnnoDF, paste0(annot_dir, outFile1), sep = "\t", quote = F, row.names = F)
}


### Enhancer-associated peaks annotation
annotatePeaks(peak = paste0(fullPeak_dir, "dunnart_enhancer_peaks.narrowPeak"), outFile = "dunnart_enhancer_annotation.txt", backg = unlist(fread("output/annotations/go_background_orthologousENSEMBL_biomart.txt", header = FALSE), use.names = FALSE),
  outFile1 = "dunnart_enhancer_annotationConvertedIDs.txt", outFile2 = "dunnart_enhancer_annotationConvertedIDs_t.devil.txt")

### Promoter-associated peaks annotation
annotatePeaks(peak = paste0(fullPeak_dir, "dunnart_promoter_peaks.narrowPeak"), outFile = "dunnart_promoter_annotation.txt", backg = unlist(fread("output/annotations/go_background_orthologousENSEMBL_biomart.txt", header = FALSE), use.names = FALSE),
  outFile1 = "dunnart_promoter_annotationConvertedIDs.txt", outFile2 = "dunnart_promoter_annotationConvertedIDs_t.devil.txt")

```

## Distance to nearest TSS
Now see where the peaks are located in relation to the TSS. Promoters should be reasonably close to the TSS and enhancers more distal to the TSS. 
Plot distance to TSS for all unfiltered peaks 

```{r}
files =list.files(annot_dir, pattern= "*.5_enhancer_annotation.txt|*.5_promoter_annotation.txt|dunnart_promoter_annotationConvertedIDs.txt|dunnart_enhancer_annotationConvertedIDs.txt", full.names=T) # create list of files in directory
filenames <- sub('\\_annotation.txt$', '', basename(files)) 
filenames <- sub('\\_annotationConvertedIDs.txt$', '', basename(filenames)) 

files = as.list(files)
data = lapply(files, function(x) fread(x, header=TRUE, sep="\t", quote = "", na.strings=c("", "NA"))) # read in all files
names(data) <- filenames


df1 = Map(mutate, data[c(1,3,5,7,9,11,13)], cre = "enhancer")
df2 = Map(mutate, data[c(2,4,6,8,10,12,14)], cre = "promoter")
data = append(df1, df2)

df1 = Map(mutate, data[c(1,8)], group = "dunnart")
df2 = Map(mutate, data[c(2,9)], group = "E10.5")
df3 = Map(mutate, data[c(3,10)], group = "E11.5")
df4 = Map(mutate, data[c(4,11)], group = "E12.5")
df5 = Map(mutate, data[c(5,12)], group = "E13.5")
df6 = Map(mutate, data[c(6,13)], group = "E14.5")
df7 = Map(mutate, data[c(7,14)], group = "E15.5")

data <- append(df1, df2)
data <- append(data, df3)
data = append(data, df4)
data = append(data, df5)
data = append(data, df6)
data = append(data, df7)

data_add_distance = suppressWarnings(lapply(data, function(x) x[,log10_abs_dist:=log10(abs(distanceToTSS)+1)][,log10_abs_dist:=ifelse(distanceToTSS<0,-log10_abs_dist,log10_abs_dist)]))

data_subset = lapply(data_add_distance, function(x) x %>% select(cre, group, log10_abs_dist) )
data_bind = rbindlist(data_subset,)
enhancers <- data_bind[data_bind$cre == "enhancer"]
promoters <- data_bind[data_bind$cre == "promoter"]

p <- ggplot(enhancers, aes(x=log10_abs_dist, color = group)) + 
  geom_density() + scale_color_manual(values = c("#fbb03b", rep("#ffd166",7))) + scale_fill_manual(values = c("#fbb03b", rep("#ffd166",7))) + theme_bw() 
p + ggtitle("enhancer distance to nearest TSS")

pdf(file = paste0(plot_dir, "mouse_dunnart_enhancer_distTSS.pdf"), width=10, height = 7)
print(p + font)
dev.off()

p <- ggplot(promoters, aes(x=log10_abs_dist, color = group)) + 
  geom_density() + scale_color_manual(values = c("#1a6259", rep("#2a9d8f",7))) + scale_fill_manual(values = c("#1a6259", rep("#2a9d8f",7))) + theme_bw()
p + ggtitle("promoter distance to nearest TSS")

pdf(file = paste0(plot_dir, "mouse_dunnart_promoter_distTSS.pdf"), width=10, height = 7)
print(p + font)
dev.off()
    
```

# k-means clustering
From this we can see that the promoter peaks have a large amount of peaks a long way from the TSS
Suggests that these are either actually enhancers or they represent unannotated transcripts. Probably a mixture of both based on comparison with mouse peaks (where the annotation is better) there are not as many peaks in this distal regions. 

Use k-means clustering to group the peaks to decide on a cut off for "promoter" peaks. This will be more conservative for what we identify as promoters.

```{r}
set.seed(123)
file = paste0(annot_dir, "dunnart_promoter_annotationConvertedIDs.txt")
plot1 = paste0(plot_dir, "dunnart_promoter_kmeans_bar.pdf")
plot2 = paste0(plot_dir, "dunnart_promoter_kmeans_histogram.pdf")
output = "dunnart_promoter_kmeans_peaks.txt"

data = fread(file, header=TRUE, sep="\t", quote = "") # read in all files
data = data[,log10_dist:=log10(abs(distanceToTSS)+1.1)][,log10_dist:=ifelse(distanceToTSS<0,-log10_dist,log10_dist)]
data = data[,abs_dist:=log10(abs(distanceToTSS)+1.1)]

data = data %>% dplyr::select("V4", "width", "V7", "distanceToTSS", "log10_dist", "abs_dist", "annotation")
  
## plot the number of peaks in each cluster
## Using the MacQueen algorithm instead of the default Lloyd 
## The algorithm is more efficient as it updates centroids more often and usually needs to
## perform one complete pass through the cases to converge on a solution.
df = data[,5]
cre.kmeans = kmeans(df, 3, iter.max=100, nstart=25, algorithm="MacQueen")  
cre.kmeans_table = data.frame(cre.kmeans$size, cre.kmeans$centers)
cre.kmeans_df = data.frame(Cluster = cre.kmeans$cluster, data)

p <- ggplot(data = cre.kmeans_df, aes(y = Cluster, 
                                      fill=as.factor(Cluster), 
                                      color=as.factor(Cluster))) +
  geom_bar()  + theme(plot.title = element_text(hjust = 0.5)) + 
  theme_bw() + scale_color_manual(values = c('#9EBCDA','#8C6BB1', "#4D004B")) + 
  scale_fill_manual(values = c('#9EBCDA','#8C6BB1', "#4D004B")) + theme_bw() 
p + ggtitle("Number of peaks in clusters")

pdf(plot1, width=10, height = 7)
print(p + font)
dev.off()
  
p <- ggplot(cre.kmeans_df, aes(x=log10_dist, 
                               fill=as.factor(Cluster), 
                               color=as.factor(Cluster))) +
  geom_histogram(binwidth=0.1, position = 'identity') +
  theme_bw() + scale_color_manual(values = c('#9EBCDA','#8C6BB1', "#4D004B")) + 
  scale_fill_manual(values = c('#9EBCDA','#8C6BB1', "#4D004B")) 
p + ggtitle("Histogram of clustered peaks")
  
pdf(plot2, width = 10, height = 7)
  print(p + font)
dev.off() 

write.table(cre.kmeans_df, paste0(filterPeaks_dir, output), sep="\t", quote=F, row.names=F)             
```


## Extract cluster groups from narrowPeak files and save separately

```{r}
cluster1 <- cre.kmeans_df$V4[cre.kmeans_df$Cluster==1]
cluster2 <- cre.kmeans_df$V4[cre.kmeans_df$Cluster==2]
cluster3 <- cre.kmeans_df$V4[cre.kmeans_df$Cluster==3]

promoter = paste0(fullPeak_dir, "dunnart_promoter_peaks.narrowPeak")
promoter_annot = paste0(annot_dir, "dunnart_promoter_annotationConvertedIDs.txt")

out1 = paste0(filterPeaks_dir, "cluster1_dunnart_promoter_peaks.narrowPeak")
out2 = paste0(filterPeaks_dir, "cluster2_dunnart_promoter_peaks.narrowPeak")
out3 =  paste0(filterPeaks_dir, "cluster3_dunnart_promoter_peaks.narrowPeak")
out4 = paste0(annot_dir, "dunnart_promoter_cluster1_annotationConvertedIDs.txt")

file= fread(promoter, header=FALSE, sep="\t", quote = "") 
file2 = fread(promoter_annot, header=TRUE, sep = "\t", quote="")
write.table(subset(file, V4 %in% cluster1), out1, quote=FALSE, col.names=FALSE, row.names=FALSE, sep="\t")
write.table(subset(file, V4 %in% cluster2), out2, quote=FALSE, col.names=FALSE, row.names=FALSE, sep="\t")
write.table(subset(file, V4 %in% cluster3), out3, quote=FALSE, col.names=FALSE, row.names=FALSE, sep="\t")
write.table(subset(file2, V4 %in% cluster1), out4, quote=FALSE, col.names=TRUE, row.names=FALSE, sep="\t")

```

```{r echo = FALSE}
message("all promoter peaks: ", dim(file)[1])
message("cluster 1 promoter peaks: ", dim(fread(paste0(filterPeaks_dir, "cluster1_dunnart_promoter_peaks.narrowPeak"), header=FALSE, sep="\t", quote = ""))[1])
message("cluster 2 promoter peaks: ", dim(fread(paste0(filterPeaks_dir, "cluster2_dunnart_promoter_peaks.narrowPeak"), header=FALSE, sep="\t", quote = ""))[1])
message("cluster 3 promoter peaks: ", dim(fread(paste0(filterPeaks_dir, "cluster3_dunnart_promoter_peaks.narrowPeak"), header=FALSE, sep="\t", quote = ""))[1])
```

## Filter mouse peaks two the same range as the dunnart clusters
### Filter peak annotation files
```{r}
files =list.files(annot_dir, pattern= ".5_promoter_annotation.txt", full.names=T) # create list of files in directory
files = as.list(files)
data_annot = lapply(files, function(x) fread(x, header=TRUE, sep="\t", quote = "", na.strings=c("", "NA"))) # read in all files
names(data_annot) = c("E10","E11","E12","E13","E14", "E15")
df1 = Map(mutate, data_annot, group = names(data_annot))

## add thresholds from kmeans clustering for cluster 1 in the dunnart
cluster1 <- fread(paste0(annot_dir, "dunnart_promoter_cluster1_annotationConvertedIDs.txt"), header=TRUE, sep="\t", quote = "")
colnames(cluster1) <- c("seqnames", "start", "end", "width", "strand", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "annotation", "geneChr", "geneStart", "geneEnd", "geneLength", "geneStrand", "geneId", "transcriptId", "distanceToTSS", "ensemblgeneID", "transcriptIdAltered", "refseqID", "combined", "mouseensembl")

range(cluster1$distanceToTSS)

## filter annotation files
df1 = lapply(df1, function(x) x %>% filter(x$distanceToTSS<200 & x$distanceToTSS>-200))

lapply(names(df1), function(x) write.table(df1[[x]], file=paste0(annot_dir,x,"_cluster1_annotation", ".txt"), sep="\t", quote=FALSE, col.names=TRUE, row.names=FALSE))
```

### Filter narrowPeak files

```{r}
## filter peak files
files = list.files(fullPeak_dir, pattern = "*.5_promoter_peaks.narrowPeak", full.names=T)
files = as.list(files)
data_peak = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA"))) # read in all files
names(data_peak) = c("E10","E11","E12","E13","E14", "E15")
df2 = Map(mutate, data_peak, group = names(data_peak))

df3 = lapply(df1, function(x) x %>% select("V4"))
df3 = lapply(df3, function(x) as.vector(x$V4))

filtered <- lapply(df2, function(x) x[x$V4 %in% unlist(df3),])


lapply(names(filtered), function(x) write.table(filtered[[x]], file=paste0(filterPeaks_dir,x,"_cluster1_peaks", ".narrowPeak"), sep="\t", quote=FALSE, col.names=FALSE, row.names=FALSE))

```

## Assess peak features after k-means clustering
### Prepare data
```{r}
files =list.files(annot_dir, pattern= "dunnart_promoter_cluster1_annotationConvertedIDs.txt|dunnart_enhancer_annotationConvertedIDs.txt|*_cluster1_annotation.txt|*.5_enhancer_annotation.txt", full.names=T) # create list of files in directory
files = as.list(files)
data = lapply(files, function(x) fread(x, header=TRUE, sep="\t", quote="", na.strings=c("", "NA")))


names(data) = c("dunnart_enhancer", "dunnart_promoter", "E10_promoter", "E10_enhancer", "E11_promoter", "E11_enhancer", "E12_promoter", "E12_enhancer", "E13_promoter", "E13_enhancer", "E14_promoter", "E14_enhancer", "E15_promoter", "E15_enhancer")


df1 = Map(mutate, data[c(2,3,5,7,9,11,13)], cre = "promoter")
df2 = Map(mutate, data[c(1,4,6,8,10,12,14)], cre = "enhancer")
data = append(df1, df2)

df1 = Map(mutate, data[c(1,8)], group = "dunnart")
df2 = Map(mutate, data[c(2,9)], group = "E10.5")
df3 = Map(mutate, data[c(3,10)], group = "E11.5")
df4 = Map(mutate, data[c(4,11)], group = "E12.5")
df5 = Map(mutate, data[c(5,12)], group = "E13.5")
df6 = Map(mutate, data[c(6,13)], group = "E14.5")
df7 = Map(mutate, data[c(7,14)], group = "E15.5")

data <- append(df1, df2)
data <- append(data, df3)
data = append(data, df4)
data = append(data, df5)
data = append(data, df6)
data = append(data, df7)

df = lapply(data, function(x) x=setnames(x, old="geneId", new="mouseensembl", skip_absent=TRUE) %>% as.data.table())
df = lapply(df, function(x) x %>% dplyr::select("width", "V7", "annotation", "mouseensembl", "distanceToTSS", "group", "cre") %>% as.data.table())

peaks = rbindlist(df,)
```

### Plot peak intensity
```{r eval=FALSE}
p <- ggplot(peaks, aes(factor(group), y = log10(V7))) + 
  geom_violin(aes(fill=factor(cre), color=factor(cre)), position = "dodge") + 
  geom_boxplot(aes(color=factor(cre)),
               width = .15, outlier.shape = NA,
               fill=c("#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec",
                      "#fcf8ec","#fcf8ec","#d3bfd2","#d3bfd2","#d3bfd2",
                      "#d3bfd2","#d3bfd2",
                      "#d3bfd2","#d3bfd2"),
               position = position_dodge(width=.1), 
               notch=TRUE) + 
  facet_wrap(. ~ cre, strip.position = "top") +
  xlab("") + 
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("Log10 Peak Intensity") + scale_color_manual(values = c("#e9c46a","#4d004b")) + 
  scale_fill_manual(values = c("#f1daa2", "#7a4078"))

p + ggtitle("Log10 Peak Intensity for enhancer-associated and high-confidence promoter-associated peaks")
    
pdf(paste0(plot_dir, "mouse_dunnart_clustered_peak_intensity.pdf"), width=10, height = 6)
print(p + font)
dev.off()
```

### Plot peak lengths
```{r}
p <- ggplot(peaks, aes(factor(group), y = log10(width))) + 
  geom_violin(aes(fill=factor(cre), color=factor(cre)), position = "dodge") + 
  geom_boxplot(aes(color=factor(cre)),
               width = .15, outlier.shape = NA,
               fill=c("#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec",
                      "#fcf8ec","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2",
                      "#d3bfd2","#d3bfd2"),
               position = position_dodge(width=.1), 
               notch=TRUE) + 
  facet_wrap(. ~ cre, strip.position = "top") +
  xlab("") + 
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("Log10 Peak Intensity") + scale_color_manual(values = c("#e9c46a","#4d004b")) + 
  scale_fill_manual(values = c("#f1daa2", "#7a4078"))

p + ggtitle("Log10 Peak Width for enhancer-associated and high-confidence promoter-associated peaks")
    
pdf(paste0(plot_dir, "mouse_dunnart_clustered_peak_width.pdf"), width=10, height = 6)
print(p + font)
dev.off()
```

### Distance to the nearest TSS
```{r}
peaks = peaks[,log10_abs_dist:=log10(abs(distanceToTSS)+1)][,log10_abs_dist:=ifelse(distanceToTSS<0,-log10_abs_dist,log10_abs_dist)]

peaks <- split(peaks, by="cre")

## enhancer is the same as above so just replot promoter
p <- ggplot(peaks$promoter, aes(x=log10_abs_dist, color = group)) + 
  geom_density() + scale_color_manual(values = c("#7a4078", rep("#ac88ab",7))) + scale_fill_manual(values = c("#7a4078", rep("#ac88ab",7))) + theme_bw()
p + ggtitle("high confidence promoter peaks - distance to nearest TSS")

pdf(file = paste0(plot_dir, "mouse_dunnart_promoter_clustered_distTSS.pdf"), width=10, height = 7)
print(p + font)
dev.off()
    
```

# Now look at CpG% and GC% for these groups
Homer can be used to look at percentage CpG and GC in enhancer/promoter sequences.

```{bash eval = FALSE}
# installed HOMER with the following modules loaded

module load gcc
module load perl
module load web_proxy
module load wget

## GC content of promoters and enhancers using homer

annotatePeaks.pl output/filtered_peaks/cluster1_dunnart_promoter_peaks.narrowPeak smiCra1 -gff data/genomic_data/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr2.gff -CpG -cons > output/annotations/cluster1_dunnart_promoter_peaks_homerAnnot.txt


## run from directory: output/peaks/ (for enhancers) and output/filtered_peaks/ (for promoters)
TRA=($(for file in E1*_cluster1_peaks.narrowPeak; do echo $file |cut -d "_" -f 1;done))

echo ${TRA[@]}

for tr in ${TRA[@]};

do
echo ${tr}

annotatePeaks.pl ${tr}_cluster1_peaks.narrowPeak mm10 -CpG > ../annotations/E10.5_promoter_cluster1_peaks_homerAnnot.txt

annotatePeaks.pl ${tr}_enhancer_peaks.narrowPeak mm10 -CpG > ../annotations/${tr}_enhancer_peaks_homerAnnot.txt

done

```

## Plot CpG and GC content across groups

```{r}

files =list.files(annot_dir, pattern= "1_peaks_homerAnnot.txt|dunnart_promoter_peaks_homerAnnot.txt|enhancer_peaks_homerAnnot.txt|enhancers_homerAnnot.txt", full.names=T) # create list of files in directory
filenames <- sub('\\_homerAnnot.txt$', '', basename(files)) 
filenames <- sub('\\_peaks$', '', basename(filenames)) 

files = as.list(files)
data = lapply(files, function(x) fread(x, header=TRUE, sep="\t", quote = "", na.strings=c("", "NA"))) # read in all files
names(data) = filenames

df1 = Map(mutate, data[c(1,3,5,7,9,11,13)], cre = "promoter")
df2 = Map(mutate, data[c(2,4,6,8,10,12,14)], cre = "enhancer")
data = append(df1, df2)

df1 = Map(mutate, data[c(1,8)], group = "dunnart")
df2 = Map(mutate, data[c(2,9)], group = "E10.5")
df3 = Map(mutate, data[c(3,10)], group = "E11.5")
df4 = Map(mutate, data[c(4,11)], group = "E12.5")
df5 = Map(mutate, data[c(5,12)], group = "E13.5")
df6 = Map(mutate, data[c(6,13)], group = "E14.5")
df7 = Map(mutate, data[c(7,14)], group = "E15.5")

data <- append(df1, df2)
data <- append(data, df3)
data = append(data, df4)
data = append(data, df5)
data = append(data, df6)
data = append(data, df7)

data <- lapply(data, function(x) x %>% select("CpG%", "GC%", "cre", "group"))

data = rbindlist(data,)

p <- ggplot(data, aes(factor(group), y = `CpG%`)) + 
  geom_violin(aes(fill=factor(cre), color=factor(cre)), position = "dodge") + 
  geom_boxplot(aes(color=factor(cre)),
               width = .15, outlier.shape = NA,
               fill=c("#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec",
                      "#fcf8ec","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2",
                      "#d3bfd2","#d3bfd2"),
               position = position_dodge(width=.1), 
               notch=TRUE) + 
  facet_wrap(. ~ cre, strip.position = "top") +
  xlab("") + 
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("CpG content") + scale_color_manual(values = c("#e9c46a","#4d004b")) + 
  scale_fill_manual(values = c("#f1daa2", "#7a4078"))

p + ggtitle("CpG content for enhancer-associated and high-confidence promoter-associated peaks")

pdf(paste0(plot_dir, "mouse_dunnart_clustered_peak_CpG_content.pdf"), width=10, height = 6)
print(p + font)
dev.off()

p <- ggplot(data, aes(factor(group), y = `GC%`)) + 
  geom_violin(aes(fill=factor(cre), color=factor(cre)), position = "dodge") + 
  geom_boxplot(aes(color=factor(cre)),
               width = .15, outlier.shape = NA,
               fill=c("#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec","#fcf8ec",
                      "#fcf8ec","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2","#d3bfd2",
                      "#d3bfd2","#d3bfd2"),
               position = position_dodge(width=.1), 
               notch=TRUE) + 
  facet_wrap(. ~ cre, strip.position = "top") +
  xlab("") + 
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ylab("GC content") + scale_color_manual(values = c("#e9c46a","#4d004b")) + 
  scale_fill_manual(values = c("#f1daa2", "#7a4078"))

p + ggtitle("GC content for enhancer-associated and high-confidence promoter-associated peaks")
    
pdf(paste0(plot_dir, "mouse_dunnart_clustered_peak_GC_content.pdf"), width=10, height = 6)
print(p + font)
dev.off()

```
