---
title: "mouse_dunnart_data_processing_for_comparison"
author: "lecook"
date: "2022-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Pipeline for mouse and dunnart peak calling


## Normalise dunnart and mouse BAM files to 10M reads 

This is so we can compare to mouse peaks.

Based on https://davemcg.github.io/post/easy-bam-downsampling/ script for doing this. 

Run script in the directory with the files you want to subsample.

```{bash eval=FALSE}
cd output/bam_files/
bash subsample.sh
```

## Read QC with deepTools

### Plot correlation between BAM files
`plotCorrelation` computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions.

This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.

```{r}
include_graphics("output/qc/pearsoncor_multibamsum_dunnart_downSampled.png")
include_graphics("output/qc/pearsoncor_multibamsum_dunnart_downSampled.png")
```

### Fingerprint plots 
This tool samples indexed BAM files and plots a profile of cumulative read coverages for each. All reads overlapping a window (bin) of the specified length are counted; these counts are sorted and the cumulative sum is finally plotted.

It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample.

An ideal input with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin etc.) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.

```{r}
include_graphics("output/qc/multiBAM_fingerprint.png")
```

### BAM PE fragment size

This tool calculates the fragment sizes for read pairs given a BAM file from paired-end sequencing. Several regions are sampled depending on the size of the genome and number of processors to estimate thesummary statistics on the fragment lengths.

```{r}
include_graphics("output/qc/bamPEFragmentSize_hist.png")
```

