---
title: "Processing mouse and dunnart data for comparison"
author: "lecook"
date: "2022-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Pipeline for mouse and dunnart peak calling

## Download mouse unfiltered alignments from ENCODE
See [mouse_data_ENCODE](mouse_data_ENCODE.html) for details on these samples.

```{bash eval=FALSE}
xargs -L 1 curl -O -J -L < ENCODE_files.txt
xargs -L 1 curl -O -J -L < control_files.txt
```

## Filter mouse alignments
Uses samtools and picards MarkDuplicates to remove low quality reads. This is based on and in accordance with the ENCODE Guidelines and pipeline (https://github.com/ENCODE-DCC/chip-seq-pipeline2)

See full snakemake script for details: code/mouse_peak_calling/mouse_H3K4me3 and code/mouse_peak_calling/mouse_H3K27ac



```{r echo=FALSE, warning = FALSE, message = FALSE}
library(data.table)
library(dplyr)
library(stringr)
library(purrr)
library(tidyverse)
library(knitr)

flagstats <- list.files("output/qc", pattern = "*dedup.flagstat.qc|*PPq30.flagstat.qc|*unfiltered.flagstat.qc|*q30.flagstat.qc|*downSampled.flagstat.qc", full.names = T) # create list of files in directory
filenames <- sub('\\.flagstat.qc$', '', basename(flagstats)) 
files <- as.list(flagstats)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))
names(data) <- filenames

stats <- lapply(data, function(x) x[1, ])
stats <- lapply(stats, function(x) data.table(read_count = substr(x, 1, 9)))
stats <- Map(dplyr::mutate, stats, file = names(stats))
stats <- rbindlist(stats,)
names <- str_split_fixed(stats$file, "\\.[A-Za-z]", 2)
stats2 <- as.data.table(cbind(stats$read_count, names))

colnames(stats2) <- c("num_reads", "file", "processing")
stats2[stats2$processing == "edup"]$processing <- "duplicates removed" 
stats2[stats2$processing == "nfiltered"]$processing <- "mapped reads" 
stats2[stats2$processing == "Pq30"]$processing <- "remove orphan reads and MAPQ < 30" 
stats2[stats2$processing == "30"]$processing <- "remove orphan reads and MAPQ < 30" 
stats2[stats2$processing == "ownSampled"]$processing <- "subsampled 10 million reads" 

setcolorder(stats2, c("file", "processing", "num_reads"))

# remove + from num_reads
stats2[stats2$processing == "subsampled 10 million reads"]$num_reads <- data.table(substr(stats2[stats2$processing == "subsampled 10 million reads"]$num_reads, 1,8))

kable(stats2, caption="Table. Read alignment quality metrics before and after filtering steps")
```

## Library complexity

__Library Complexity ChIP-seq Standards:__

| PBC1 | PBC2 | Bottlenecking level | NRF | Complexity | Flag colors |
|:-:|:-:|:-:|:-:|:-:|:-:|
| < 0.5 | < 1 | Severe | < 0.5 | Concerning | Orange |
| 0.5 ≤ PBC1 < 0.8 | 1 ≤ PBC2 < 3 | Moderate | 0.5 ≤ NRF < 0.8 | Acceptable | Yellow |
| 0.8 ≤ PBC1 < 0.9 | 3 ≤ PBC2 < 10 | Mild | 0.8 ≤ NRF < 0.9 | Compliant | None |
| ≥ 0.9 | ≥ 10 | None | > 0.9 | Ideal | None |


```{r echo=FALSE, warning = FALSE, message = FALSE}

lib.complex <- list.files("output/qc", pattern = "*pbc.qc", full.names = T) # create list of files in directory
filenames <- sub('\\.pbc.qc$', '', basename(lib.complex)) 
files <- as.list(lib.complex)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))

names(data) <- filenames
stats <- Map(dplyr::mutate, data, file = names(data))
stats <- rbindlist(stats,)

colnames(stats) = c("TotalReadPairs", "DistinctReadPairs", 
                "OneReadPair", "TwoReadPairs",
                "NRF=Distinct/Total", "PBC1=OnePair/Distinct",
                "PBC2=OnePair/TwoPair", "file")
setcolorder(stats, c("file", "TotalReadPairs", "DistinctReadPairs", 
                "OneReadPair", "TwoReadPairs",
                "NRF=Distinct/Total", "PBC1=OnePair/Distinct",
                "PBC2=OnePair/TwoPair"))

kable(stats, caption="Table. Library complexity metrics")
```

## Normalise dunnart and mouse BAM files to 10M reads 

This is so we can compare to mouse peaks.

Based on https://davemcg.github.io/post/easy-bam-downsampling/ script for doing this. 

Run script in the directory with the files you want to subsample.

```{bash eval=FALSE}
cd output/bam_files/
bash subsample.sh
```

## Read QC with deepTools

### Plot correlation between BAM files
`plotCorrelation` computes the overall similarity between two or more files based on read coverage (or other scores) within genomic regions.

This helps to determine whether the different sample types can be separated, i.e., samples of different conditions are expected to be more dissimilar to each other than replicates within the same condition.

![Pearson correlation dunnart downsampled BAM files](pearsoncor_multibamsum_dunnart_downSampled.pdf){height=500px width=600px}
![Pearson correlation mouse downsampled BAM files for H3K4me3](H3K4me3_pearsoncor_multibamsum_mouse.pdf){height=500px width=600px}
![Pearson correlation mouse downsampled BAM files for H3K27ac](H3K27ac_pearsoncor_multibamsum_mouse.pdf){height=500px width=600px}

### Fingerprint plots 
This tool samples indexed BAM files and plots a profile of cumulative read coverages for each. All reads overlapping a window (bin) of the specified length are counted; these counts are sorted and the cumulative sum is finally plotted.

It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample.

An ideal input with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin etc.) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.

![Fingerprint plot for dunnart downsampled BAM files](multiBAM_fingerprint_dunnart_downSampled.pdf){height=500px width=600px}
![Fingerprint plot for mouse downsampled BAM files for H3K4me3](H3K4me3_multiBAM_fingerprint_mouse.pdf){height=500px width=600px}
![Fingerprint plot for mouse downsampled BAM files for H3K27ac](H3K27ac_multiBAM_fingerprint_mouse.pdf){height=500px width=600px}


## Cross-correlation analysis
Information from: https://docs.google.com/document/d/1lG_Rd7fnYgRpSIqrIfuVlAz2dW1VaSQThzk836Db99c/edit

This set of programs operate on mapped Illumina single-end read datasets in tagAlign or BAM format. Because my data is paired-end I need to only use the forward read.

A high-quality ChIP-seq experiment will produce significant clustering of enriched DNA sequence tags/reads at locations bound by the protein of interest; the expectation is that we can observe a bimodal enrichment of reads (sequence tags) on both the forward and the reverse strands.

Cross-correlation analysis is done on a filtered (but not-deduped) and subsampled BAM. There is a special fastq trimming for cross-correlation analysis. Read1 fastq is trimmed to 50bp first using trimfastq.py (last modified 2017/11/08, https://github.com/ENCODE-DCC/chip-seq-pipeline2/blob/master/src/trimfastq.py). And then it is separately mapped as SE. Reads are filtered but duplicates are not removed. Then 15 million reads are randomly sampled and used for cross-correlation analysis.

ENCODE standards:

| col. | abbreviation     | description                                                                                         |
|------|------------------|-----------------------------------------------------------------------------------------------------|
| 1    | Filename         | tagAlign/BAM filename                                                                               |
| 2    | numReads         | effective sequencing depth i.e. total number of mapped reads in input file                          |
| 3    | estFragLen       | comma separated strand cross-correlation peak(s) in decreasing order of correlation.                |
| 4    | corr_estFragLen  | comma separated strand cross-correlation value(s) in decreasing order (COL2 follows the same order) |
| 5    | phantomPeak      | Read length/phantom peak strand shift                                                               |
| 6    | corr_phantomPeak | Correlation value at phantom peak                                                                   |
| 7    | argmin_corr      | strand shift at which cross-correlation is lowest                                                   |
| 8    | min_corr         | minimum value of cross-correlation                                                                  |
| 9    | NSC              | Normalized strand cross-correlation coefficient (NSC) = COL4 / COL8                                 |
| 10   | RSC              | Relative strand cross-correlation coefficient (RSC) = (COL4 - COL8) / (COL6 - COL8)                 |
| 11   | QualityTag       | Quality tag based on thresholded RSC (codes= -2:veryLow, -1:Low, 0:Medium, 1:High, 2:veryHigh)      |

NSC; NSC>1.1 (higher values indicate more enrichment; 1 = no enrichment)

RSC; RSC>0.8 (0 = no signal; <1 low quality ChIP; >1 high enrichment

Quality tag based on thresholded RSC (codes: -2:veryLow,-1:Low,0:Medium,1:High; 2:veryHigh)


```{r echo=FALSE, warning = FALSE, message = FALSE}

ssp <- list.files("output/qc", pattern = "*cc.qc", full.names = T) # create list of files in directory
filenames <- sub('\\_filt_15Mreads.SE.cc.qc$', '', basename(ssp)) 
files <- as.list(ssp)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))

names(data) <- filenames
stats <- Map(dplyr::mutate, data, file = names(data))
stats <- rbindlist(stats,)
stats <- stats[,-1]
colnames(stats) = c("numReads", "estFragLen", 
                "corr_estFragLen", "PhantomPeak",
                "corr_phantomPeak", "argmin_corr",
                "min_corr", "phantomPeakCoef", "relPhantomPeakCoef",
                "QualityTag", "file")
setcolorder(stats, c("file", "numReads", "estFragLen", 
                "corr_estFragLen", "PhantomPeak",
                "corr_phantomPeak", "argmin_corr",
                "min_corr", "phantomPeakCoef", "relPhantomPeakCoef",
                "QualityTag"))

kable(stats, caption="Table. Peak calling quality metrics for mouse and dunnart ChIP-seq peaks after peak calling with MACS2")
```

## Peak QC
### Number of peaks in read normalised data (10 million reads) 

#### Dunnart
```{bash echo = FALSE}
echo "Number of peaks A3 H3K27ac dunnart default"
less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks A2 H3K4me3 dunnart default"
less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks B3 H3K27ac dunnart default"
less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks B2 H3K4me3 dunnart default"
less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_default_dunnart_downSampled_peaks.narrowPeak | wc -l

echo "Number of peaks A3 H3K27ac dunnart P < 0.01"
less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_p0.01_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks A2 H3K4me3 dunnart P < 0.01"
less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_p0.01_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks B3 H3K27ac dunnart P < 0.01"
less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_p0.01_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks B2 H3K4me3 dunnart P < 0.01"
less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_p0.01_dunnart_downSampled_peaks.narrowPeak | wc -l
```

#### Mouse
H3K4me3
```{bash echo = FALSE}
echo "Number of peaks rep 1 E10.5 H3K4me3"
less output/peaks/ENCFF124UYX_vs_ENCFF157KEH_E10.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E10.5 H3K4me3"
less output/peaks/ENCFF045IPK_vs_ENCFF825AVI_E10.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E11.5 H3K4me3"
less output/peaks/ENCFF760QYZ_vs_ENCFF184CUE_E11.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E11.5 H3K4me3"
less output/peaks/ENCFF717QDV_vs_ENCFF376FGM_E11.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E12.5 H3K4me3"
less output/peaks/ENCFF941QJZ_vs_ENCFF058AUT_E12.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E12.5 H3K4me3"
less output/peaks/ENCFF182ZPF_vs_ENCFF203JQV_E12.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E13.5 H3K4me3"
less output/peaks/ENCFF485UDC_vs_ENCFF117QRC_E13.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E13.5 H3K4me3"
less output/peaks/ENCFF124TAB_vs_ENCFF248PGK_E13.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E14.5 H3K4me3"
less output/peaks/ENCFF665QBJ_vs_ENCFF002HZV_E14.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E14.5 H3K4me3"
less output/peaks/ENCFF724DMU_vs_ENCFF784ORI_E14.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E15.5 H3K4me3"
less output/peaks/ENCFF401BKM_vs_ENCFF182XFG_E15.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E15.5 H3K4me3"
less output/peaks/ENCFF258KCR_vs_ENCFF727QTS_E15.5_H3K4me3_macs2_downSampled_peaks.narrowPeak | wc -l
```

H3K27ac
```{bash echo = FALSE}
echo "Number of peaks rep 1 E10.5 H3K27ac"
less output/peaks/ENCFF213EBC_vs_ENCFF157KEH_E10.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E10.5 H3K27ac"
less output/peaks/ENCFF548BRR_vs_ENCFF825AVI_E10.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E11.5 H3K27ac"
less output/peaks/ENCFF512SFE_vs_ENCFF184CUE_E11.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E11.5 H3K27ac"
less output/peaks/ENCFF515PKL_vs_ENCFF376FGM_E11.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E12.5 H3K27ac"
less output/peaks/ENCFF011NFM_vs_ENCFF058AUT_E12.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E12.5 H3K27ac"
less output/peaks/ENCFF394TZN_vs_ENCFF203JQV_E12.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E13.5 H3K27ac"
less output/peaks/ENCFF194ORC_vs_ENCFF117QRC_E13.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E13.5 H3K27ac"
less output/peaks/ENCFF290ZNF_vs_ENCFF248PGK_E13.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E14.5 H3K27ac"
less output/peaks/ENCFF902HAR_vs_ENCFF002HZV_E14.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l 
echo "Number of peaks rep 2 E14.5 H3K27ac"
less output/peaks/ENCFF327VAO_vs_ENCFF784ORI_E14.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E15.5 H3K27ac"
less output/peaks/ENCFF707WKL_vs_ENCFF182XFG_E15.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l 
echo "Number of peaks rep 2 E15.5 H3K27ac"
less output/peaks/ENCFF584JFB_vs_ENCFF727QTS_E15.5_H3K27ac_macs2_downSampled_peaks.narrowPeak | wc -l
```

### Fraction of reads in peaks 

```{r echo=FALSE, warning = FALSE, message = FALSE}

files <- list.files("output/qc", pattern = "*.frip_default_dunnart_downSampled.txt|*downSampled.frip.txt", full.names = T) # create list of files in directory
filenames <- sub('\\.txt$', '', basename(files)) 
files <- as.list(files)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))
names(data) <- filenames

stats <- lapply(data, function(x) data.table(t(x)))
stats <- lapply(stats, function(x) data.table(reads_in_peaks = substr(x$V1, 45, 55), total_reads =substr(x$V2, 24,35), fraction_reads_in_peaks = substr(x$V3, 30,36)))

stats <- Map(dplyr::mutate, stats, peak_file = names(stats))
stats <- rbindlist(stats,)

stats$peak_file[1:4] <- substr(str_subset(stats$peak_file,"dunnart"), 1,24)
stats$peak_file[5:28] <- substr(str_subset(stats$peak_file,"frip"), 1,40)

setcolorder(stats, c("peak_file", "total_reads", "reads_in_peaks", "fraction_reads_in_peaks"))


kable(stats, caption="Table. Fraction of reads in peaks for replicate peak files")
```

### Number of peaks for all reads 
#### Dunnart
```{bash echo = FALSE}
echo "Number of peaks A3 H3K27ac dunnart default"
less output/peaks/A-3_H3K27ac_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l
echo "Number of peaks A2 H3K4me3 dunnart default"
less output/peaks/A-2_H3K4me3_vs_A-1_input_macs2_default_peaks.narrowPeak | wc -l
echo "Number of peaks B3 H3K27ac dunnart default"
less output/peaks/B-3_H3K27ac_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l
echo "Number of peaks B2 H3K4me3 dunnart default"
less output/peaks/B-2_H3K4me3_vs_B-1_input_macs2_default_peaks.narrowPeak | wc -l
```

#### Mouse
H3K4me3
```{bash echo = FALSE}
echo "Number of peaks rep 1 E10.5 H3K4me3"
less output/peaks/ENCFF124UYX_vs_ENCFF157KEH_E10.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E10.5 H3K4me3"
less output/peaks/ENCFF045IPK_vs_ENCFF825AVI_E10.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E11.5 H3K4me3"
less output/peaks/ENCFF760QYZ_vs_ENCFF184CUE_E11.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E11.5 H3K4me3"
less output/peaks/ENCFF717QDV_vs_ENCFF376FGM_E11.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E12.5 H3K4me3"
less output/peaks/ENCFF941QJZ_vs_ENCFF058AUT_E12.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E12.5 H3K4me3"
less output/peaks/ENCFF182ZPF_vs_ENCFF203JQV_E12.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E13.5 H3K4me3"
less output/peaks/ENCFF485UDC_vs_ENCFF117QRC_E13.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E13.5 H3K4me3"
less output/peaks/ENCFF124TAB_vs_ENCFF248PGK_E13.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E14.5 H3K4me3"
less output/peaks/ENCFF665QBJ_vs_ENCFF002HZV_E14.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E14.5 H3K4me3"
less output/peaks/ENCFF724DMU_vs_ENCFF784ORI_E14.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E15.5 H3K4me3"
less output/peaks/ENCFF401BKM_vs_ENCFF182XFG_E15.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E15.5 H3K4me3"
less output/peaks/ENCFF258KCR_vs_ENCFF727QTS_E15.5_H3K4me3_macs2_peaks.narrowPeak | wc -l
```

H3K27ac
```{bash echo = FALSE}
echo "Number of peaks rep 1 E10.5 H3K27ac"
less output/peaks/ENCFF213EBC_vs_ENCFF157KEH_E10.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E10.5 H3K27ac"
less output/peaks/ENCFF548BRR_vs_ENCFF825AVI_E10.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E11.5 H3K27ac"
less output/peaks/ENCFF512SFE_vs_ENCFF184CUE_E11.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E11.5 H3K27ac"
less output/peaks/ENCFF515PKL_vs_ENCFF376FGM_E11.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E12.5 H3K27ac"
less output/peaks/ENCFF011NFM_vs_ENCFF058AUT_E12.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E12.5 H3K27ac"
less output/peaks/ENCFF394TZN_vs_ENCFF203JQV_E12.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E13.5 H3K27ac"
less output/peaks/ENCFF194ORC_vs_ENCFF117QRC_E13.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 2 E13.5 H3K27ac"
less output/peaks/ENCFF290ZNF_vs_ENCFF248PGK_E13.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E14.5 H3K27ac"
less output/peaks/ENCFF902HAR_vs_ENCFF002HZV_E14.5_H3K27ac_macs2_peaks.narrowPeak | wc -l 
echo "Number of peaks rep 2 E14.5 H3K27ac"
less output/peaks/ENCFF327VAO_vs_ENCFF784ORI_E14.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
echo "Number of peaks rep 1 E15.5 H3K27ac"
less output/peaks/ENCFF707WKL_vs_ENCFF182XFG_E15.5_H3K27ac_macs2_peaks.narrowPeak | wc -l 
echo "Number of peaks rep 2 E15.5 H3K27ac"
less output/peaks/ENCFF584JFB_vs_ENCFF727QTS_E15.5_H3K27ac_macs2_peaks.narrowPeak | wc -l
```

### Fraction of reads in peaks 

```{r echo=FALSE, warning = FALSE, message = FALSE}

files <- list.files("output/qc", pattern = "*.frip_default.txt|*3.frip.txt|*c.frip.txt", full.names = T) # create list of files in directory
filenames <- sub('\\.txt$', '', basename(files)) 
files <- as.list(files)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))
names(data) <- filenames

stats <- lapply(data, function(x) data.table(t(x)))
stats <- lapply(stats, function(x) data.table(reads_in_peaks = substr(x$V1, 45, 55), total_reads =substr(x$V2, 24,35), fraction_reads_in_peaks = substr(x$V3, 30,36)))

stats <- Map(dplyr::mutate, stats, peak_file = names(stats))
stats <- rbindlist(stats,)

stats$peak_file[1:4] <- substr(str_subset(stats$peak_file,"input"), 1,24)
stats$peak_file[5:28] <- substr(str_subset(stats$peak_file,"frip"), 1,40)

setcolorder(stats, c("peak_file", "total_reads", "reads_in_peaks", "fraction_reads_in_peaks"))


kable(stats, caption="Table. Fraction of reads in peaks for replicate peak files")
```


### Consensus peaksets for biological replicates (normalised read count)
#### Number of consensus peaks 
##### Dunnart
```{bash echo = FALSE}
echo "Number of consensus H3K4me3 dunnart peaks"
less output/peaks/H3K4me3_overlap_default_dunnart_downSampled_peaks.narrowPeak | wc -l
echo "Number of consensus H3K27ac dunnart peaks"
less output/peaks/H3K27ac_overlap_default_dunnart_downSampled_peaks.narrowPeak | wc -l
```

##### Mouse
```{bash echo = FALSE}
echo "Number of consensus H3K4me3 mouse E10.5 peaks"
less output/peaks/H3K4me3_E10.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E11.5 peaks"
less output/peaks/H3K4me3_E11.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E12.5 peaks"
less output/peaks/H3K4me3_E12.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E13.5 peaks"
less output/peaks/H3K4me3_E13.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E14.5 peaks"
less output/peaks/H3K4me3_E14.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E15.5 peaks"
less output/peaks/H3K4me3_E15.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E10.5 peaks"
less output/peaks/H3K27ac_E10.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E11.5 peaks"
less output/peaks/H3K27ac_E11.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E12.5 peaks"
less output/peaks/H3K27ac_E12.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E13.5 peaks"
less output/peaks/H3K27ac_E13.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E14.5 peaks"
less output/peaks/H3K27ac_E14.5_downSampled_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E15.5 peaks"
less output/peaks/H3K27ac_E15.5_downSampled_overlap.narrowPeak | wc -l
```

#### Fraction of reads in peaks for consensus peaks

```{r echo=FALSE, warning = FALSE, message = FALSE, eval=FALSE}
files <- list.files("output/qc", pattern = "*default_dunnart_downSampled.frip|*overlap_downSampled.frip", full.names = T) # create list of files in directory
filenames <- sub('\\.frip$', '', basename(files)) 
files <- as.list(files)
dunnart_downSampled
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))
names(data) <- filenames

stats <- lapply(data, function(x) data.table(t(x)))
stats <- lapply(stats, function(x) data.table(reads_in_peaks = substr(x$V1, 45, 55), total_reads =substr(x$V2, 24,35), fraction_reads_in_peaks = substr(x$V3, 30,36)))

stats <- Map(dplyr::mutate, stats, peak_file = names(stats))
stats <- rbindlist(stats,)

stats$peak_file[c(1:6, 8:13)] <- substr(str_subset(stats$peak_file,".5_overlap"), 1,13)
stats$peak_file[c(7,14)] <- paste(substr(str_subset(stats$peak_file,"downSampled"), 1, 7), 
                                  substr(str_subset(stats$peak_file,"downSampled"), 25, 31), 
                                  sep="_")

setcolorder(stats, c("peak_file", "total_reads", "reads_in_peaks", "fraction_reads_in_peaks"))


kable(stats, caption="Table. Fraction of reads in peaks for replicate peak files")
```


### Consensus peaksets for biological replicates (all reads)
#### Number of consensus peaks 
##### Dunnart
```{bash echo = FALSE}
echo "Number of consensus H3K4me3 dunnart peaks"
less output/peaks/H3K4me3_overlap_default.narrowPeak | wc -l
echo "Number of consensus H3K27ac dunnart peaks"
less output/peaks/H3K27ac_overlap_default.narrowPeak | wc -l
```

H3K4me3_overlap_default.narrowPeak

##### Mouse
```{bash echo = FALSE}
echo "Number of consensus H3K4me3 mouse E10.5 peaks"
less output/peaks/H3K4me3_E10.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E11.5 peaks"
less output/peaks/H3K4me3_E11.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E12.5 peaks"
less output/peaks/H3K4me3_E12.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E13.5 peaks"
less output/peaks/H3K4me3_E13.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E14.5 peaks"
less output/peaks/H3K4me3_E14.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K4me3 mouse E15.5 peaks"
less output/peaks/H3K4me3_E15.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E10.5 peaks"
less output/peaks/H3K27ac_E10.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E11.5 peaks"
less output/peaks/H3K27ac_E11.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E12.5 peaks"
less output/peaks/H3K27ac_E12.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E13.5 peaks"
less output/peaks/H3K27ac_E13.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E14.5 peaks"
less output/peaks/H3K27ac_E14.5_overlap.narrowPeak | wc -l
echo "Number of consensus H3K27ac mouse E15.5 peaks"
less output/peaks/H3K27ac_E15.5_overlap.narrowPeak | wc -l
```

#### Fraction of reads in peaks for consensus peaks

```{r echo=FALSE, warning = FALSE, message = FALSE}
files <- list.files("output/qc", pattern = "*default.frip|*overlap.frip", full.names = T) # create list of files in directory
filenames <- sub('\\.frip$', '', basename(files)) 
files <- as.list(files)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))
names(data) <- filenames

stats <- lapply(data, function(x) data.table(t(x)))
stats <- lapply(stats, function(x) data.table(reads_in_peaks = substr(x$V1, 45, 55), total_reads =substr(x$V2, 24,35), fraction_reads_in_peaks = substr(x$V3, 30,36)))

stats <- Map(dplyr::mutate, stats, peak_file = names(stats))
stats <- rbindlist(stats,)

stats$peak_file[c(1:6, 8:13)] <- substr(str_subset(stats$peak_file,".5_overlap"), 1,13)
stats$peak_file[c(7,14)] <- paste(substr(str_subset(stats$peak_file,"overlap")
                                         , 1, 7), "dunnart", 
                                  sep="_")

setcolorder(stats, c("peak_file", "total_reads", "reads_in_peaks", "fraction_reads_in_peaks"))


kable(stats, caption="Table. Fraction of reads in peaks for replicate peak files")
```

# Grouping peaks into putative enhancers and promoters

Find uniquely H3K4me3 sites (i.e. peaks that don't overlap with H3K27ac):

```{bash eval=FALSE}
conda activate chip
bedtools intersect -v -a output/peaks/H3K4me3_E10.5_overlap.narrowPeak -b output/peaks/H3K27ac_E10.5_overlap.narrowPeak > output/peaks/H3K4me3_E10.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E11.5_overlap.narrowPeak -b output/peaks/H3K27ac_E11.5_overlap.narrowPeak > output/peaks/H3K4me3_E11.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E12.5_overlap.narrowPeak -b output/peaks/H3K27ac_E12.5_overlap.narrowPeak > output/peaks/H3K4me3_E12.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E13.5_overlap.narrowPeak -b output/peaks/H3K27ac_E13.5_overlap.narrowPeak > output/peaks/H3K4me3_E13.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E14.5_overlap.narrowPeak -b output/peaks/H3K27ac_E14.5_overlap.narrowPeak > output/peaks/H3K4me3_E14.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K4me3_E15.5_overlap.narrowPeak -b output/peaks/H3K27ac_E15.5_overlap.narrowPeak > output/peaks/H3K4me3_E15.5_only.narrowPeak
```

Find unique H3K27ac sites (i.e. peaks that don't overlap with H3K4me3):

```{bash eval=FALSE}
bedtools intersect -v -a output/peaks/H3K27ac_E10.5_overlap.narrowPeak -b output/peaks/H3K4me3_E10.5_overlap.narrowPeak > output/peaks/H3K27ac_E10.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E11.5_overlap.narrowPeak -b output/peaks/H3K4me3_E11.5_overlap.narrowPeak > output/peaks/H3K27ac_E11.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E12.5_overlap.narrowPeak -b output/peaks/H3K4me3_E12.5_overlap.narrowPeak > output/peaks/H3K27ac_E12.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E13.5_overlap.narrowPeak -b output/peaks/H3K4me3_E13.5_overlap.narrowPeak > output/peaks/H3K27ac_E13.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E14.5_overlap.narrowPeak -b output/peaks/H3K4me3_E14.5_overlap.narrowPeak > output/peaks/H3K27ac_E14.5_only.narrowPeak
bedtools intersect -v -a output/peaks/H3K27ac_E15.5_overlap.narrowPeak -b output/peaks/H3K4me3_E15.5_overlap.narrowPeak > output/peaks/H3K27ac_E15.5_only.narrowPeak

```

Find peaks common between H3K27ac & H3K4me3 with a reciprocal overlap of at least 50%.
```{bash eval=FALSE}
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E10.5_overlap.narrowPeak -b output/peaks/H3K27ac_E10.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E10.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E11.5_overlap.narrowPeak -b output/peaks/H3K27ac_E11.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E11.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E12.5_overlap.narrowPeak -b output/peaks/H3K27ac_E12.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E12.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E13.5_overlap.narrowPeak -b output/peaks/H3K27ac_E13.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E13.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E14.5_overlap.narrowPeak -b output/peaks/H3K27ac_E14.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E14.5.narrowPeak
bedtools intersect -f 0.5 -r -a output/peaks/H3K4me3_E15.5_overlap.narrowPeak -b output/peaks/H3K27ac_E15.5_overlap.narrowPeak > output/peaks/H3K4me3_and_H3K27ac_E15.5.narrowPeak

```

Combine peak files

```{bash}
cat output/peaks/H3K4me3_E10.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E10.5.narrowPeak | sort -k 4,4 > output/peaks/E10.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E11.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E11.5.narrowPeak | sort -k 4,4 > output/peaks/E11.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E12.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E12.5.narrowPeak | sort -k 4,4 > output/peaks/E12.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E13.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E13.5.narrowPeak | sort -k 4,4 > output/peaks/E13.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E14.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E14.5.narrowPeak | sort -k 4,4 > output/peaks/E14.5_promoter_peaks.narrowPeak
cat output/peaks/H3K4me3_E15.5_only.narrowPeak output/peaks/H3K4me3_and_H3K27ac_E15.5.narrowPeak | sort -k 4,4 > output/peaks/E15.5_promoter_peaks.narrowPeak

cat output/peaks/H3K27ac_E10.5_only.narrowPeak | sort -k 4,4 > output/peaks/E10.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E11.5_only.narrowPeak | sort -k 4,4 > output/peaks/E11.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E12.5_only.narrowPeak | sort -k 4,4 > output/peaks/E12.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E13.5_only.narrowPeak | sort -k 4,4 > output/peaks/E13.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E14.5_only.narrowPeak | sort -k 4,4 > output/peaks/E14.5_enhancer_peaks.narrowPeak
cat output/peaks/H3K27ac_E15.5_only.narrowPeak | sort -k 4,4 > output/peaks/E15.5_enhancer_peaks.narrowPeak
```

