---
title: "Cross-species peak level comparisons"
author: "lecook"
date: "2022-02-23"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include = FALSE} 
knitr::opts_chunk$set(warning = FALSE, message = FALSE) 
knitr::opts_chunk$set(echo = TRUE)

library(data.table)
library(knitr)
library(dplyr)
library(stringr)
library(purrr)
library(tidyverse)
library(GenomicFeatures)
library(ChIPseeker)


plot_dir <- "output/plots/"
fullPeak_dir <- "output/peaks/"
annot_dir <- "output/annotations/"
filterPeaks_dir <- "output/filtered_peaks/"
liftover_dir <- "output/liftover/"

smiCra_txdb <- makeTxDbFromGFF("data/genomic_data/Scras_dunnart_assem1.0_pb-ont-illsr_flyeassem_red-rd-scfitr2_pil2xwgs2_60chr2.gff")
## geneID conversion tables
df2 <- read.table("output/annotations/devil_to_mouse_ensembl.txt", header=TRUE, sep="\t") ## conversion table for devil ENSEMBL to mouse ENSEMBL
df3 <- read.table("output/annotations/refseq_to_ensembl.txt", header=TRUE, sep="\t") ## convertsion table for devil refseq to devil ENSEMBL
bg = fread("output/annotations/go_background_orthologousENSEMBL_biomart.txt", header = FALSE)

```

# Summary tables
After the tables is all the code to compare peaks between species using liftOver and bedtools. It is a lot of repetitive code that I could revise but it was easiest to track it all this way for now. The summary tables show the number of peaks for the different overlaps.

## Number of peaks lifted over with no width filtering
```{r echo=FALSE}
files <- list.files(liftover_dir, pattern = "dunnart_enhancer_smiCraTOmm10.bed|dunnart_promoter_smiCraTOmm10.bed", full.names = T)
files <- as.list(files)
data <- lapply(files, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA")))

df <- data.table(enhancer_peaks_smiCraTOmm10 = dim(data[[1]])[1], promoter_peak_smiCraTOmm10 = dim(data[[2]])[1])

kable(df, caption="Table. Number of peaks lifted over from dunnart to mouse")
```

## Liftover peak summits from dunnart to mouse
```{r echo=FALSE, warning = FALSE, message = FALSE}

summarise_liftover <- function(file_pattern, column_name, files){
fl <- list.files(liftover_dir, pattern = file_pattern, full.names = T) # create list of files in directory
fl <- fl[{files}]
fl <- as.list(fl)
data <- lapply(fl, function(x) data.table::fread(x, header = FALSE, sep = "\t", quote = "", na.strings = c("", "NA"), fill=TRUE))
df <- lapply(data, function(x) data.table(dim(x)[1]))
df <- lapply(df, function(x) {x=setnames(x,column_name)})
return(df)
}

compare_nearest_genes_after_liftover <- function(files, i, j){
reciprocal.to.dunnart = list.files(liftover_dir, pattern= "*annotation_smiCraTOmm10_mm10TOsmiCra.bed", full.names=T)
reciprocal.to.dunnart <- reciprocal.to.dunnart[{files}]
reciprocal.to.dunnart <- as.list(reciprocal.to.dunnart)

# annotate lifted over coordinates
reciprocal.to.dunnart.anno <- lapply(reciprocal.to.dunnart, annotatePeak, tssRegion = c(-3000, 3000), TxDb = smiCra_txdb, verbose=FALSE)

## create list of dataframe annotations
reciprocal.to.dunnart.annoList = list(reciprocal.to.dunnart.anno[[1]]@anno, reciprocal.to.dunnart.anno[[2]]@anno,
                        reciprocal.to.dunnart.anno[[3]]@anno, reciprocal.to.dunnart.anno[[4]]@anno,
                        reciprocal.to.dunnart.anno[[5]]@anno, reciprocal.to.dunnart.anno[[6]]@anno)
peakAnnoDF <- lapply(reciprocal.to.dunnart.annoList, function(x) as.data.table(x))
        
        
# Convert new annotations (after liftOver) to ensembl IDs in the mouse
peakAnnoDF = lapply(peakAnnoDF, function(x){
  x <- x[
    ,ensemblgeneID:=df2$Gene.stable.ID[match(unlist(x$geneId), df2$Gene.name)]
    ][
      ,ensemblgeneID:=replace(x$ensemblgeneID,is.na(x$ensemblgeneID),"-")
      ][
        ,transcriptIdAltered:=gsub("\\..*","", x$transcriptId)
        ][
          ,refseqID:=df3$Ensembl.Gene.ID[match(unlist(x$transcriptIdAltered), df3$RefSeq.mRNA.Accession)]
          ][
            ,refseqID:=replace(x$refseqID,is.na(x$refseqID),"-")
            ][
              ,combined:=ifelse(x$refseqID == "-", x$ensemblgeneID, x$refseqID)
              ]
})

length.ensembl = lapply(peakAnnoDF, function(x) length(which(x$ensemblgeneID %like% "ENS" == TRUE)))
length.refseq = lapply(peakAnnoDF, function(x) length(x$transcriptId))

peakAnnoDF = lapply(peakAnnoDF, function(x) x[,refseq_score:=ifelse(x[[i]] ==transcriptId & transcriptId %like% "XM_", "1", "0")])
peakAnnoDF = lapply(peakAnnoDF, function(x) x[,ensembl_score:=ifelse(combined == x[[j]] & x[[j]] %like% "ENS", "1", "0")])

gene.comparison.refseq.counts = lapply(peakAnnoDF, function(x) data.table(length(which(x$refseq_score ==1))))   

gene.comparison.ensembl.counts = lapply(peakAnnoDF, function(x) data.table("number of alignable peaks with the same nearest gene call before and after liftOver" = length(which(x$ensembl_score ==1))))

return(gene.comparison.ensembl.counts)
}


## Promoter-associated peaks
dunnart_to_mouse_promoters = rbindlist(summarise_liftover(file_pattern = "*bpsummits_smiCraTOmm10.bed", column_name = "liftOver dunnart to mouse", files = 7:12),)

mouse_to_dunnart_promoters = rbindlist(summarise_liftover(file_pattern = "*.bed_smiCraTOmm10_mm10TOsmiCra.bed", column_name = "liftOver mouse to dunnart (reciprocal)", files = 7:12), )

ensembl_alignable_promoters = rbindlist(summarise_liftover(file_pattern = "*bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed", column_name = "alignable peaks with ENSEMBL annotations", files = 7:12), )

compare_nearest_genes_promoters = rbindlist(compare_nearest_genes_after_liftover(7:12, i = "V26", j = "V31"),)

promoter_peaks = cbind(dunnart_to_mouse_promoters, mouse_to_dunnart_promoters, ensembl_alignable_promoters, compare_nearest_genes_promoters)

promoter_peaks <- promoter_peaks[
    ,summit_width:=c("100bp", "200bp", "300bp", "400bp", "500bp", "50bp")
    ][
      ,`number of dunnart peaks`:= rep("8510", 6)
      ][
        ,`% alignable peaks of total peaks`:=(as.numeric(`liftOver dunnart to mouse`) 
                                              / as.numeric(`number of dunnart peaks`))*100
        ][
          ,`% of alignable peaks with the same gene call before and after liftOver`:= 
            (as.numeric(`number of alignable peaks with the same nearest gene call before and after liftOver`/
                          as.numeric(`alignable peaks with ENSEMBL annotations`)))*100
          ][
            c(6,1,2,3,4,5),
            ][
              ,c(5,6,1,2,7,3,4,8)
              ]

## Enhancer-associated peaks 
dunnart_to_mouse_enhancers = rbindlist(summarise_liftover(file_pattern = "*bpsummits_smiCraTOmm10.bed", column_name = "liftOver dunnart to mouse", files = 1:6),)

mouse_to_dunnart_enhancers = rbindlist(summarise_liftover(file_pattern = "*.bed_smiCraTOmm10_mm10TOsmiCra.bed", column_name = "liftOver mouse to dunnart (reciprocal)", files = 1:6),)

ensembl_alignable_enhancers = rbindlist(summarise_liftover(file_pattern = "*bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed", column_name = "alignable peaks with ENSEMBL annotations", files = 1:6), )

compare_nearest_genes_enhancers = rbindlist(compare_nearest_genes_after_liftover(1:6, i = "V24", j = "V29"),)

enhancer_peaks <- cbind(dunnart_to_mouse_enhancers, mouse_to_dunnart_enhancers, ensembl_alignable_enhancers, compare_nearest_genes_enhancers)

enhancer_peaks <- enhancer_peaks[
    ,summit_width:=c("100bp", "200bp", "300bp", "400bp", "500bp", "50bp")
    ][
      ,`number of dunnart peaks`:= rep("35676", 6)
      ][
        ,`% alignable peaks of total peaks`:=(as.numeric(`liftOver dunnart to mouse`) 
                                              / as.numeric(`number of dunnart peaks`))*100
        ][
          ,`% of alignable peaks with the same gene call before and after liftOver`:= 
            (as.numeric(`number of alignable peaks with the same nearest gene call before and after liftOver`/
                          as.numeric(`alignable peaks with ENSEMBL annotations`)))*100
          ][
            c(6,1,2,3,4,5),
            ][
              ,c(5,6,1,2,7,3,4,8)
              ]


kable(promoter_peaks, caption="Summary of alignable promoter-associated peaks in the mouse and dunnart")
kable(enhancer_peaks, caption="Summary of alignable enhancer-associated peaks in the mouse and dunnart")

```

## Summary of mouse-dunnart conserved a. high-confidence promoter-associated peaks,
and b. enhancer-associated peaks.
```{r eval=FALSE}
# filter to peaks that lift Over back to the same nearest gene
promoters.filtered = lapply(promoters, function(x) x[x$ensembl_score == 1])
enhancers.filtered = lapply(enhancers, function(x) x[x$ensembl_score == 1])

# Convert devil ensembl to mouse ensembl
promoters.filtered = lapply(promoters.filtered, function(x) x[,mouseensembl:=df2$Mouse.gene.stable.ID[match(unlist(x$combined), df2$Gene.stable.ID)]])
promoters.filtered = lapply(promoters.filtered, function(x) x %>% dplyr::select(seqnames, start, end, width, geneId, transcriptId, distanceToTSS, refseq_score, ensembl_score, mouseensembl))
promoters.filtered[[6]][promoters.filtered[[6]] == ""] <- NA
write.table(promoters.filtered[[6]], file = "promoters.filtered.liftOver.50bpsummits.annotations.bed", sep="\t", quote=F, row.names=F, col.names = F)
enhancers.filtered = lapply(enhancers.filtered, function(x) x[,mouseensembl:=df2$Mouse.gene.stable.ID[match(unlist(x$combined), df2$Gene.stable.ID)]])
enhancers.filtered = lapply(enhancers.filtered, function(x) x %>% dplyr::select(seqnames, start, end, width, geneId, transcriptId, distanceToTSS, refseq_score, ensembl_score, mouseensembl))
enhancers.filtered[[6]][enhancers.filtered[[6]] == ""] <- NA
write.table(enhancers.filtered[[6]], file = "enhancers.filtered.liftOver.50bpsummits.annotations.bed", sep="\t", quote=F, row.names=F, col.names = F)


```

# LiftOver of peak coordinates between species
## All peaks with no peak width filtering

```{bash eval=FALSE}
## Load modules
conda activate chip
module load ucsc

## Extract peak coordinates
less output/peaks/dunnart_enhancer_peaks.narrowPeak | awk '{print $1,$2,$3}' > output/peaks/dunnart_enhancer_peaks_coord.bed
less output/filtered_peaks/cluster1_dunnart_promoter_peaks.narrowPeak | awk '{print $1,$2,$3}' > output/filtered_peaks/dunnart_promoter_peaks_coord.bed

## LiftOver peak coordinates from dunnart to mouse
liftOver output/peaks/dunnart_enhancer_peaks_coord.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_smiCraTOmm10.bed output/liftover/dunnart_enhancer_smiCraTOmm10_unmapped.bed
liftOver output/filtered_peaks/dunnart_promoter_peaks_coord.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_smiCraTOmm10.bed output/liftover/dunnart_promoter_smiCraTOmm10_unmapped.bed
```

<!-- # Import output files in R -->
<!-- enh = fread("enhancers/dunnart.enhancer_smiCraTOmm10.bed") -->
<!-- prom = fread("promoters/clustered/dunnart.promoter_smiCraTOmm10.bed") -->
<!-- data = list(enh, prom) -->
<!-- names(data) = c("enhancers", "promoters") -->
<!-- data = Map(mutate, data, cre = names(data)) -->
<!-- data = lapply(data, function(x) x[order(V4),]) -->
<!-- data = lapply(data, function(x) x=x[,width:=V3-V2]) -->
<!-- data = rbindlist(data,) -->

## LiftOver peak summits
### Prepare peak summits and extract peak IDs
```{bash eval=FALSE}

## Extract peak IDs
less output/peaks/dunnart_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}' | sort > output/peaks/dunnart_enhancer_peaks_IDs.txt
less output/filtered_peaks/cluster1_dunnart_promoter_peaks.narrowPeak | awk -F '\t' '{print $4}' | sort > output/filtered_peaks/dunnart_promoter_peaks_IDs.txt

less output/peaks/E10.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E10.5_enhancer_peaks_IDs.txt
less  output/peaks/E11.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E11.5_enhancer_peaks_IDs.txt
less  output/peaks/E12.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E12.5_enhancer_peaks_IDs.txt
less  output/peaks/E13.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E13.5_enhancer_peaks_IDs.txt
less  output/peaks/E14.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E14.5_enhancer_peaks_IDs.txt
less  output/peaks/E15.5_enhancer_peaks.narrowPeak | awk -F '\t' '{print $4}'  >  output/peaks/E15.5_enhancer_peaks_IDs.txt

less  output/filtered_peaks/E10_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}' > output/filtered_peaks/E10.5_promoter_peaks_IDs.txt
less  output/filtered_peaks/E11_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}'  > output/filtered_peaks/E11.5_promoter_peaks_IDs.txt
less  output/filtered_peaks/E12_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}'  > output/filtered_peaks/E12.5_promoter_peaks_IDs.txt
less  output/filtered_peaks/E13_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}'  > output/filtered_peaks/E13.5_promoter_peaks_IDs.txt
less  output/filtered_peaks/E14_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}'  > output/filtered_peaks/E14.5_promoter_peaks_IDs.txt
less  output/filtered_peaks/E15_cluster1_peaks.narrowPeak | awk -F '\t' '{print $4}'  > output/filtered_peaks/E15.5_promoter_peaks_IDs.txt

## Concatenate pooled summit peak files 

cat output/peaks/H3K4me3_pooled_macs2_default_summits.bed output/peaks/H3K27ac_pooled_macs2_default_summits.bed > output/peaks/H3K4me3_H3K27ac_dunnart_pooled_default_combinedSummits.bed

cat output/peaks/E10.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E10.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E10.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed
cat output/peaks/E11.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E11.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E11.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed
cat output/peaks/E12.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E12.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E12.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed
cat output/peaks/E13.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E13.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E13.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed
cat output/peaks/E14.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E14.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E14.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed
cat output/peaks/E15.5_H3K27ac.pooled.macs2_summits.bed output/peaks/E15.5_H3K4me3.pooled.macs2_summits.bed > output/peaks/E15.5_H3K27ac_H3K4me3.pooled.macs2_combinedSummits.bed

```

### Extract rows from summits file based on enhancer and promoter peak IDs
```{r eval=FALSE}

files =list.files(fullPeak_dir, pattern= "macs2_combinedSummits.bed|default_combinedSummits.bed", full.names=T) # create list of files in directory
files = as.list(files)
summits = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA")))
names(summits) <- c("E10", "E11", "E12", "E13", "E14", "E15", "dunnart")

files2 =list.files(fullPeak_dir, pattern= "peaks_IDs.txt", full.names=T) # create list of files in directory
files3 = list.files(filterPeaks_dir, pattern = "peaks_IDs.txt", full.names=T)
files = append(files2, files3)
filenames <- sub('\\_peaks_IDs.txt$', '', basename(files)) 

files = as.list(files)

peak_IDs = lapply(files, function(x) fread(x, header=FALSE, sep="\t", quote = "", na.strings=c("", "NA")))
names(peak_IDs) <- filenames

filtered <- lapply(summits, function(x) x[x$V4 %in% unlist(as.vector(peak_IDs)),])
filtered <- lapply(filtered, function(x) substr(x$V4, 1, 13))

filtered_bind <- rbindlist(filtered,)
filtered_bind$V4 <- substr(filtered_bind$V4, 1,13)
filtered_split <- split(filtered_bind, by = "V4")

names(filtered_split) <- c("E10.5_enhancer", "E10.5_promoter", 
                           "E11.5_enhancer", "E11.5_promoter",
                           "E12.5_enhancer", "E12.5_promoter",
                           "E13.5_enhancer", "E13.5_promoter",
                           "E14.5_enhancer", "E14.5_promoter",
                           "E15.5_enhancer", "E15.5_promoter",
                           "dunnart_promoter", "dunnart_enhancer")

lapply(names(filtered_split), function(x) write.table(filtered_split[[x]], file=paste0("output/liftover/",x,"_summits", ".bed"), sep="\t", quote=FALSE, col.names=FALSE, row.names=FALSE))
```

### Extend the summits 25bp in both directions

```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 25 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_50bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 25 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_50bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_50bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_50bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_50bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_50bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_50bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_50bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_50bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_50bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_50bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_50bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_50bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 25 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_50bpsummits.bed
```

### Extend the summits 50bp in both directions
```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 50 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_100bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 50 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_100bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_100bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_100bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_100bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_100bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_100bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_100bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_100bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_100bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_100bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_100bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_100bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 50 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_100bpsummits.bed
```

### Extend the summits 100bp in both directions
```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 100 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_200bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 100 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_200bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_200bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_200bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_200bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_200bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_200bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_200bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_200bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_200bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_200bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_200bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_200bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 100 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_200bpsummits.bed
```

### Extend the summits 150bp in both directions
```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 150 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_300bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 150 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_300bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_300bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_300bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_300bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_300bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_300bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_300bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_300bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_300bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_300bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_300bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_300bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 150 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_300bpsummits.bed
```

### Extend the summits 200bp in both directions
```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 200 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_400bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 200 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_400bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_400bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_400bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_400bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_400bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_400bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_400bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_400bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_400bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_400bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_400bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_400bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 200 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_400bpsummits.bed
```

### Extend the summits 250bp in both directions
```{bash eval=FALSE}
bedtools slop -i output/liftover/dunnart_promoter_summits.bed -b 250 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_promoter_500bpsummits.bed

bedtools slop -i output/liftover/dunnart_enhancer_summits.bed -b 250 -g data/genomic_data/smiCra1.chrom.sizes > output/liftover/dunnart_enhancer_500bpsummits.bed

bedtools slop -i output/liftover/E10.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_enhancer_500bpsummits.bed
bedtools slop -i output/liftover/E11.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_enhancer_500bpsummits.bed
bedtools slop -i output/liftover/E12.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_enhancer_500bpsummits.bed
bedtools slop -i output/liftover/E13.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_enhancer_500bpsummits.bed
bedtools slop -i output/liftover/E14.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_enhancer_500bpsummits.bed
bedtools slop -i output/liftover/E15.5_enhancer_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_enhancer_500bpsummits.bed

bedtools slop -i output/liftover/E10.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E10.5_promoter_500bpsummits.bed
bedtools slop -i output/liftover/E11.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E11.5_promoter_500bpsummits.bed
bedtools slop -i output/liftover/E12.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E12.5_promoter_500bpsummits.bed
bedtools slop -i output/liftover/E13.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E13.5_promoter_500bpsummits.bed
bedtools slop -i output/liftover/E14.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E14.5_promoter_500bpsummits.bed
bedtools slop -i output/liftover/E15.5_promoter_summits.bed -b 250 -g data/genomic_data/mm10.chrom.sizes > output/liftover/E15.5_promoter_500bpsummits.bed
```

### LiftOver extended peak summits
#### LiftOver dunnart peaks to mouse genome
```{bash eval=FALSE}

## Enhancer-associated peaks
liftOver output/liftover/dunnart_enhancer_50bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_50bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_50bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_100bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_100bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_100bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_200bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_200bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_200bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_300bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_300bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_300bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_400bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_400bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_400bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_500bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_500bpsummits_smiCraTOmm10.bed output/liftover/dunnart_enhancer_500bpsummits_smiCraTOmm10_unmapped.bed

## Promoter-associated peaks
liftOver output/liftover/dunnart_promoter_50bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_50bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_50bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_100bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_100bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_100bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_200bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_200bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_200bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_300bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_300bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_300bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_400bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_400bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_400bpsummits_smiCraTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_500bpsummits.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_500bpsummits_smiCraTOmm10.bed output/liftover/dunnart_promoter_500bpsummits_smiCraTOmm10_unmapped.bed
```

#### Lift back to the dunnart genome
```{bash eval=FALSE}
## Enhancer-associated peaks
liftOver output/liftover/dunnart_enhancer_50bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_enhancer_100bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_enhancer_200bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_enhancer_300bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_enhancer_400bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_enhancer_500bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

## Promoter-associated peaks
liftOver output/liftover/dunnart_promoter_50bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_promoter_100bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_promoter_200bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_promoter_300bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_promoter_400bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver output/liftover/dunnart_promoter_500bpsummits_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_unmapped.bed
```


#### Lift back to mouse genome so that everything is in mouse coordinates
```{bash eval=FALSE}
## Enhancer-assocaited peaks
liftOver output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

## Promoter-associated peaks 
liftOver output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

liftOver output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10_unmapped.bed

```

### Intersect dunnart CREs with mouse CREs
```{bash eval=FALSE}
## Promoter-associated peaks
bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_promoter_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_promoter_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_promoter_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_promoter_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_promoter_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_promoter_500bpsummits.bed

## Enhancer-associated peaks
bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_100bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_200bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_300bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_400bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak -wo > output/liftover/E10.5_mm10intersect_dunnart_enhancer_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak -wo > output/liftover/E11.5_mm10intersect_dunnart_enhancer_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak -wo > output/liftover/E12.5_mm10intersect_dunnart_enhancer_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak -wo > output/liftover/E13.5_mm10intersect_dunnart_enhancer_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak -wo > output/liftover/E14.5_mm10intersect_dunnart_enhancer_500bpsummits.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak -wo > output/liftover/E15.5_mm10intersect_dunnart_enhancer_500bpsummits.bed
```



### Use window to allow intersections within 50 bps 
```{bash eval=FALSE}
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E10_cluster1_peaks.narrowPeak  > output/liftover/E10.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E11_cluster1_peaks.narrowPeak  > output/liftover/E11.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E12_cluster1_peaks.narrowPeak  > output/liftover/E12.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E13_cluster1_peaks.narrowPeak  > output/liftover/E13.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E14_cluster1_peaks.narrowPeak  > output/liftover/E14.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_promoter_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/filtered_peaks/E15_cluster1_peaks.narrowPeak  > output/liftover/E15.5_mm10intersect_dunnart_promoter_50bpsummits_window.bed

bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E10.5_enhancer_peaks.narrowPeak  > output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E11.5_enhancer_peaks.narrowPeak  > output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E12.5_enhancer_peaks.narrowPeak  > output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E13.5_enhancer_peaks.narrowPeak  > output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E14.5_enhancer_peaks.narrowPeak  > output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
bedtools window -w 50 -u -a output/liftover/dunnart_enhancer_50bpsummits.bed_smiCraTOmm10_mm10TOsmiCra_backTOmm10.bed -b output/peaks/E15.5_enhancer_peaks.narrowPeak  > output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed
```

#### Lift dunnart peaks back over to the dunnart genome

```{bash eval=FALSE}
liftOver -bedPlus=4 output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E10.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E11.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E12.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E13.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E14.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra.bed output/liftover/E15.5_mm10intersect_dunnart_enhancer_50bpsummits_window_backTOsmiCra_unmapped.bed

```

## When you lift over a peak from one species to another, does it always land next to the same gene in both species?
Use annotated peak files for all peaks and compare nearest gene call

### Intersect annotation file with peak summits

```{bash eval=FALSE}
bedtools intersect -a output/liftover/dunnart_promoter_50bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_50bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_50bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_50bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_50bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_50bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_50bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_50bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_50bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_50bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_50bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_50bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_50bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_50bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_50bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_50bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_50bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_50bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_50bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_promoter_100bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_100bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_100bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_100bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_100bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_100bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_100bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_100bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_100bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_100bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_100bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_100bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_100bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_100bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_100bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_100bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_100bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_100bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_100bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_promoter_200bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_200bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_200bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_200bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_200bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_200bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_200bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_200bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_200bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_200bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_200bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_200bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_200bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_200bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_200bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_200bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_200bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_200bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_200bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_promoter_300bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_300bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_300bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_300bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_300bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_300bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_300bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_300bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_300bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_300bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_300bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_300bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_300bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_300bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_300bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_300bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_300bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_300bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_300bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_promoter_400bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_400bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_400bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_400bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_400bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_400bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_400bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_400bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_400bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_400bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_400bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_400bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_400bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_400bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_400bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_400bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_400bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_400bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_400bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_promoter_500bpsummits.bed -b output/annotations/dunnart_promoter_cluster1_annotationConvertedIDs.bed -wb > output/liftover/dunnart_promoter_500bpsummits_annotation.bed

bedtools intersect -a output/liftover/dunnart_enhancer_500bpsummits.bed -b output/annotations/dunnart_enhancer_annotationConvertedIDs.bed -wb > output/liftover/dunnart_enhancer_500bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_promoter_500bpsummits.bed -b output/annotations/E10_cluster1_annotation.bed -wb > output/liftover/E10_promoter_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_promoter_500bpsummits.bed -b output/annotations/E11_cluster1_annotation.bed -wb > output/liftover/E11_promoter_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_promoter_500bpsummits.bed -b output/annotations/E12_cluster1_annotation.bed -wb > output/liftover/E12_promoter_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_promoter_500bpsummits.bed -b output/annotations/E13_cluster1_annotation.bed -wb > output/liftover/E13_promoter_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_promoter_500bpsummits.bed -b output/annotations/E14_cluster1_annotation.bed -wb > output/liftover/E14_promoter_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_promoter_500bpsummits.bed -b output/annotations/E15_cluster1_annotation.bed -wb > output/liftover/E15_promoter_500bpsummits_annotation.bed

bedtools intersect -a output/liftover/E10.5_enhancer_500bpsummits.bed -b output/annotations/E10.5_enhancer_annotation.bed -wb > output/liftover/E10_enhancer_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E11.5_enhancer_500bpsummits.bed -b output/annotations/E11.5_enhancer_annotation.bed -wb > output/liftover/E11_enhancer_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E12.5_enhancer_500bpsummits.bed -b output/annotations/E12.5_enhancer_annotation.bed -wb > output/liftover/E12_enhancer_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E13.5_enhancer_500bpsummits.bed -b output/annotations/E13.5_enhancer_annotation.bed -wb > output/liftover/E13_enhancer_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E14.5_enhancer_500bpsummits.bed -b output/annotations/E14.5_enhancer_annotation.bed -wb > output/liftover/E14_enhancer_500bpsummits_annotation.bed
bedtools intersect -a output/liftover/E15.5_enhancer_500bpsummits.bed -b output/annotations/E15.5_enhancer_annotation.bed -wb > output/liftover/E15_enhancer_500bpsummits_annotation.bed
```


### Liftover dunnart peaks with annotation to mouse
```{bash eval=FALSE}
liftOver -bedPlus=4 output/liftover/dunnart_promoter_50bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_50bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_50bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_100bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_100bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_100bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_200bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_200bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_200bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_300bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_300bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_300bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_400bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_400bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_400bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_500bpsummits_annotation.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_500bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_promoter_500bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_50bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_50bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_50bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_100bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_100bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_100bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_200bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_200bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_200bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_300bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_300bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_300bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_400bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_400bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_400bpsummits_annotation_smiCraTOmm10_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_500bpsummits_annotation_edited.bed output/wga/liftOver_chains/smiCra1TOmm10_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_500bpsummits_annotation_smiCraTOmm10.bed output/liftover/dunnart_enhancer_500bpsummits_annotation_smiCraTOmm10_unmapped.bed
```

### Liftover peaks with annotation back to dunnart
```{bash eval=FALSE}
liftOver -bedPlus=4 output/liftover/dunnart_promoter_50bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_50bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_50bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_100bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_100bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_100bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_200bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_200bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_200bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_300bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_300bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_300bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_400bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_400bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_400bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_promoter_500bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_promoter_500bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_promoter_500bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_50bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_50bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_50bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_100bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_100bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_100bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_200bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_200bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_200bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_300bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_300bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_300bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_400bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_400bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_400bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed

liftOver -bedPlus=4 output/liftover/dunnart_enhancer_500bpsummits_annotation_smiCraTOmm10.bed output/wga/liftOver_chains/mm10TOsmiCra1_patched_sorted_repFil_chainCl_preNet_chainNet_stitched.chain output/liftover/dunnart_enhancer_500bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra.bed output/liftover/dunnart_enhancer_500bpsummits_annotation_smiCraTOmm10_mm10TOsmiCra_unmapped.bed
```
